<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 3: Modeling Part 1 | Help! What Statistical Model Should I Use? 2024</title>
  <meta name="description" content="Workshop website for the 2024 Help! What Statistical Model Should I Use? Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 3: Modeling Part 1 | Help! What Statistical Model Should I Use? 2024" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://cbw-dev.github.io/STAT_2024/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the 2024 Help! What Statistical Model Should I Use? Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="cbw-dev/STAT_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 3: Modeling Part 1 | Help! What Statistical Model Should I Use? 2024" />
  
  <meta name="twitter:description" content="Workshop website for the 2024 Help! What Statistical Model Should I Use? Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://cbw-dev.github.io/STAT_2024/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: Patrick McMillan, Alysha Cooper, Jacqueline May" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-2-dealing-with-missingness.html"/>
<link rel="next" href="module-4-modeling-part-2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Help! What Statistical Model Should I Use?<br><br></a></strong></center></li>
<a href="./">
 <img src="img/STAT_2024_icon.png" width="70%" alt="Help! What Statistical Model Should I Use? logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="meet-your-faculty.html"><a href="meet-your-faculty.html"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
<li class="chapter" data-level="" data-path="data-and-compute-setup.html"><a href="data-and-compute-setup.html"><i class="fa fa-check"></i>Data and Compute Setup</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html"><i class="fa fa-check"></i>Module 1: Data Cleaning and Exploration Review</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#lab"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#load-libraries--"><i class="fa fa-check"></i>Load Libraries —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#read-in-data--"><i class="fa fa-check"></i>Read in Data —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#looking-at-your-data.--"><i class="fa fa-check"></i>Looking at your data. —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#working-with-different-data-types.--"><i class="fa fa-check"></i>Working with different data types. —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#summary-stats.--"><i class="fa fa-check"></i>Summary stats. —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#exploratory-plots.--"><i class="fa fa-check"></i>Exploratory plots. —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#outlier-detection.--"><i class="fa fa-check"></i>Outlier detection. —-</a></li>
<li class="chapter" data-level="" data-path="module-1-data-cleaning-and-exploration-review.html"><a href="module-1-data-cleaning-and-exploration-review.html#dimensionality-reduction.--"><i class="fa fa-check"></i>Dimensionality reduction. —-</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html"><i class="fa fa-check"></i>Module 2: Dealing with Missingness</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#lab-1"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#load-libraries.--"><i class="fa fa-check"></i>Load libraries. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#read-in-data.--"><i class="fa fa-check"></i>Read in data. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#missing-data-exploration.--"><i class="fa fa-check"></i>Missing data exploration. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#imputation-prep.--"><i class="fa fa-check"></i>Imputation prep. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#missingness-simulation.--"><i class="fa fa-check"></i>Missingness simulation. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#meanmode-imputation.--"><i class="fa fa-check"></i>Mean/mode imputation. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#k-nearest-neighbour-imputation--"><i class="fa fa-check"></i>K-Nearest Neighbour Imputation —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#random-forest-imputation--"><i class="fa fa-check"></i>Random forest imputation —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#mice.--"><i class="fa fa-check"></i>MICE. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#error-rates.--"><i class="fa fa-check"></i>Error rates. —-</a></li>
<li class="chapter" data-level="" data-path="module-2-dealing-with-missingness.html"><a href="module-2-dealing-with-missingness.html#impute-the-full-dataset.--"><i class="fa fa-check"></i>Impute the full dataset. —-</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html"><i class="fa fa-check"></i>Module 3: Modeling Part 1</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#lecture-2"><i class="fa fa-check"></i>Lecture</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#regression"><i class="fa fa-check"></i>Regression</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#generalized-linear-models"><i class="fa fa-check"></i>Generalized Linear Models</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#lab-2"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#loading-the-libraries"><i class="fa fa-check"></i>Loading the Libraries</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#multiple-linear-regression-and-the-dietswap-data"><i class="fa fa-check"></i>Multiple Linear Regression and the Dietswap Data</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#logistic-regression-1"><i class="fa fa-check"></i>Logistic Regression</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#binomial-regression"><i class="fa fa-check"></i>Binomial Regression</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#poisson-regression"><i class="fa fa-check"></i>Poisson Regression</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#negative-binomial"><i class="fa fa-check"></i>Negative Binomial</a></li>
<li class="chapter" data-level="" data-path="module-3-modeling-part-1.html"><a href="module-3-modeling-part-1.html#independent-work"><i class="fa fa-check"></i>Independent Work</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-modeling-part-2.html"><a href="module-4-modeling-part-2.html"><i class="fa fa-check"></i>Module 4: Modeling Part 2</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-modeling-part-2.html"><a href="module-4-modeling-part-2.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4-modeling-part-2.html"><a href="module-4-modeling-part-2.html#lab-3"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-putting-it-all-together.html"><a href="module-5-putting-it-all-together.html"><i class="fa fa-check"></i>Module 5: Putting it all Together</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-putting-it-all-together.html"><a href="module-5-putting-it-all-together.html#lecture-4"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-5-putting-it-all-together.html"><a href="module-5-putting-it-all-together.html#lab-4"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-introduction-to-causal-intference.html"><a href="module-6-introduction-to-causal-intference.html"><i class="fa fa-check"></i>Module 6: Introduction to Causal Intference</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-introduction-to-causal-intference.html"><a href="module-6-introduction-to-causal-intference.html#lecture-5"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-6-introduction-to-causal-intference.html"><a href="module-6-introduction-to-causal-intference.html#lab-5"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>

<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=25% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
<a href="https://www.hpcforhealth.ca/"><img src="img/sponsors/hpc4health.png" width=60% style="display:inline-block;" style="padding-bottom: 20px "alt="HPC4Health logo.">
<br>
<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="70" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br>
<a href="https://computationalgenomics.ca/"><img src="img/sponsors/c3g.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Canadian Centre for Computational Genomics logo.">
<br><br>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Help! What Statistical Model Should I Use? 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-3-modeling-part-1" class="section level1 hasAnchor">
<h1>Module 3: Modeling Part 1<a href="module-3-modeling-part-1.html#module-3-modeling-part-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="lecture-2" class="section level2 hasAnchor">
<h2>Lecture<a href="module-3-modeling-part-1.html#lecture-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="regression" class="section level3 hasAnchor">
<h3>Regression<a href="module-3-modeling-part-1.html#regression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<iframe width="640" height="360" src="https://www.youtube.com/embed/pcSaS4TBzHU?si=eCHtcM7dbNW3l6w0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen>
</iframe>
</div>
<div id="generalized-linear-models" class="section level3 hasAnchor">
<h3>Generalized Linear Models<a href="module-3-modeling-part-1.html#generalized-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<iframe width="640" height="360" src="https://www.youtube.com/embed/Mk9wS0wB5Kc?si=CZEhM-EYh7cmsxgf" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen>
</iframe>
<p><br></p>
<iframe src="https://drive.google.com/file/d/1tL8V3YQ1iFUdwRQOAxqWS56yANCR_kZk/preview" width="640" height="480" allow="autoplay">
</iframe>
</div>
</div>
<div id="lab-2" class="section level2 hasAnchor">
<h2>Lab<a href="module-3-modeling-part-1.html#lab-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="loading-the-libraries" class="section level3 hasAnchor">
<h3>Loading the Libraries<a href="module-3-modeling-part-1.html#loading-the-libraries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, let’s load any necessary libraries</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="module-3-modeling-part-1.html#cb132-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># General functions and plots</span></span>
<span id="cb132-2"><a href="module-3-modeling-part-1.html#cb132-2" tabindex="-1"></a><span class="fu">library</span>(faraway) <span class="co"># Source of the gala dataset</span></span>
<span id="cb132-3"><a href="module-3-modeling-part-1.html#cb132-3" tabindex="-1"></a><span class="fu">library</span>(car) <span class="co"># For some model diagnostics</span></span>
<span id="cb132-4"><a href="module-3-modeling-part-1.html#cb132-4" tabindex="-1"></a><span class="fu">library</span>(arm) <span class="co"># For binned plot for logistic regression</span></span>
<span id="cb132-5"><a href="module-3-modeling-part-1.html#cb132-5" tabindex="-1"></a><span class="fu">library</span>(mgcv) <span class="co"># For beta regression</span></span>
<span id="cb132-6"><a href="module-3-modeling-part-1.html#cb132-6" tabindex="-1"></a><span class="fu">library</span>(microbiome) <span class="co"># Source of dietswap data</span></span>
<span id="cb132-7"><a href="module-3-modeling-part-1.html#cb132-7" tabindex="-1"></a><span class="fu">library</span>(microbiomeDataSets) <span class="co"># Source of baboongut data</span></span>
<span id="cb132-8"><a href="module-3-modeling-part-1.html#cb132-8" tabindex="-1"></a><span class="fu">library</span>(phyloseq) <span class="co"># to calculate diversity measures</span></span>
<span id="cb132-9"><a href="module-3-modeling-part-1.html#cb132-9" tabindex="-1"></a><span class="fu">library</span>(GUniFrac) <span class="co"># for throat dataset</span></span>
<span id="cb132-10"><a href="module-3-modeling-part-1.html#cb132-10" tabindex="-1"></a><span class="fu">library</span>(MGLM) <span class="co"># for rna seq dataset</span></span></code></pre></div>
<p>First, we will cover multiple linear regression modeling in R with the lm() function. In regression modeling, it is always important to consider the data available to you. What is your response of interest? Is it a continuous, roughly normally distributed variable? Which predictors should be included in your model? Do these predictors have a linear relationship with the outcome?</p>
</div>
<div id="multiple-linear-regression-and-the-dietswap-data" class="section level3 hasAnchor">
<h3>Multiple Linear Regression and the Dietswap Data<a href="module-3-modeling-part-1.html#multiple-linear-regression-and-the-dietswap-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Supposed researchers are interested in how nationality and bmi group are related to diversity of bacteria in the gut microbiome. In particular, researchers are interested to know whether nationality moderates the relationship between bmi group and Shannon diversity.</p>
<div id="loading-the-data" class="section level4 hasAnchor">
<h4>Loading the Data<a href="module-3-modeling-part-1.html#loading-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="module-3-modeling-part-1.html#cb133-1" tabindex="-1"></a><span class="co"># load the dietswap data from the microbiome package</span></span>
<span id="cb133-2"><a href="module-3-modeling-part-1.html#cb133-2" tabindex="-1"></a><span class="fu">data</span>(dietswap)</span>
<span id="cb133-3"><a href="module-3-modeling-part-1.html#cb133-3" tabindex="-1"></a></span>
<span id="cb133-4"><a href="module-3-modeling-part-1.html#cb133-4" tabindex="-1"></a><span class="co"># look up description of dataset</span></span>
<span id="cb133-5"><a href="module-3-modeling-part-1.html#cb133-5" tabindex="-1"></a>?dietswap</span>
<span id="cb133-6"><a href="module-3-modeling-part-1.html#cb133-6" tabindex="-1"></a></span>
<span id="cb133-7"><a href="module-3-modeling-part-1.html#cb133-7" tabindex="-1"></a><span class="co"># given that this is a phylogenetic class format, we get the data as follows:</span></span>
<span id="cb133-8"><a href="module-3-modeling-part-1.html#cb133-8" tabindex="-1"></a><span class="fu">otu_table</span>(dietswap) <span class="co"># counts of different OTUs</span></span>
<span id="cb133-9"><a href="module-3-modeling-part-1.html#cb133-9" tabindex="-1"></a><span class="fu">sample_data</span>(dietswap) <span class="co"># the metadata</span></span>
<span id="cb133-10"><a href="module-3-modeling-part-1.html#cb133-10" tabindex="-1"></a></span>
<span id="cb133-11"><a href="module-3-modeling-part-1.html#cb133-11" tabindex="-1"></a><span class="co"># let&#39;s look at the summary of our sample data</span></span>
<span id="cb133-12"><a href="module-3-modeling-part-1.html#cb133-12" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sample_data</span>(dietswap))</span></code></pre></div>
<p>Since we do not have a column for the Shannon diversity, we must calculate it ourselves with the estimate_richness() function from the phyloseq package.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="module-3-modeling-part-1.html#cb134-1" tabindex="-1"></a><span class="co"># we don&#39;t have shannon diversity so we must calculate it given the OTU table</span></span>
<span id="cb134-2"><a href="module-3-modeling-part-1.html#cb134-2" tabindex="-1"></a>Shannon <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(<span class="fu">otu_table</span>(dietswap), <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>))</span>
<span id="cb134-3"><a href="module-3-modeling-part-1.html#cb134-3" tabindex="-1"></a></span>
<span id="cb134-4"><a href="module-3-modeling-part-1.html#cb134-4" tabindex="-1"></a><span class="co"># add shannon diversity to our sample data</span></span>
<span id="cb134-5"><a href="module-3-modeling-part-1.html#cb134-5" tabindex="-1"></a>dietswap_data <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(<span class="at">Shannon=</span>Shannon<span class="sc">$</span>Shannon, <span class="fu">sample_data</span>(dietswap))</span>
<span id="cb134-6"><a href="module-3-modeling-part-1.html#cb134-6" tabindex="-1"></a></span>
<span id="cb134-7"><a href="module-3-modeling-part-1.html#cb134-7" tabindex="-1"></a><span class="co"># look at histogram of Shannon diversity</span></span>
<span id="cb134-8"><a href="module-3-modeling-part-1.html#cb134-8" tabindex="-1"></a><span class="fu">hist</span>(dietswap_data<span class="sc">$</span>Shannon)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="module-3-modeling-part-1.html#cb135-1" tabindex="-1"></a><span class="co"># look at boxplot of Shannon diversity</span></span>
<span id="cb135-2"><a href="module-3-modeling-part-1.html#cb135-2" tabindex="-1"></a><span class="fu">boxplot</span>(dietswap_data<span class="sc">$</span>Shannon)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-19-2.png" width="672" /></p>
<p>Another potential issue we notice when reviewing the dataset is that each sample was observed at multiple time points which means this data will violate our independence assumption. To overcome this, we can filter our data to only look at the first time point.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="module-3-modeling-part-1.html#cb136-1" tabindex="-1"></a><span class="co"># remove second time point for each group</span></span>
<span id="cb136-2"><a href="module-3-modeling-part-1.html#cb136-2" tabindex="-1"></a>dietswap_data_t1 <span class="ot">&lt;-</span> dietswap_data <span class="sc">%&gt;%</span></span>
<span id="cb136-3"><a href="module-3-modeling-part-1.html#cb136-3" tabindex="-1"></a>                  <span class="fu">filter</span>(timepoint.within.group<span class="sc">==</span><span class="dv">1</span>)</span></code></pre></div>
<p>Since the researchers are interested to know whether there is an interaction effect, we must include the interaction and main effects in our linear model. To include an interaction between two variables, we use ’*’ in the model which will automatically include both main effects and the interaction effect of the 2 variables.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="module-3-modeling-part-1.html#cb137-1" tabindex="-1"></a><span class="co"># fitting the full multiple linear regression in R</span></span>
<span id="cb137-2"><a href="module-3-modeling-part-1.html#cb137-2" tabindex="-1"></a>mod1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Shannon<span class="sc">~</span>nationality<span class="sc">*</span>bmi_group<span class="sc">+</span>sex, <span class="at">data=</span>dietswap_data_t1)</span>
<span id="cb137-3"><a href="module-3-modeling-part-1.html#cb137-3" tabindex="-1"></a><span class="fu">summary</span>(mod1)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = Shannon ~ nationality * bmi_group + sex, data = dietswap_data_t1)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.32963 -0.36395  0.09898  0.33401  1.44820 
## 
## Coefficients:
##                                    Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)                          2.7540     0.2296  11.993   &lt;2e-16 ***
## nationalityAFR                      -0.3100     0.2212  -1.401   0.1640    
## bmi_groupoverweight                  0.2642     0.2275   1.162   0.2481    
## bmi_groupobese                      -0.2012     0.2208  -0.911   0.3643    
## sexmale                             -0.0509     0.1377  -0.370   0.7125    
## nationalityAFR:bmi_groupoverweight  -0.5859     0.2934  -1.997   0.0484 *  
## nationalityAFR:bmi_groupobese       -0.0190     0.2880  -0.066   0.9475    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.5512 on 105 degrees of freedom
## Multiple R-squared:  0.2439, Adjusted R-squared:  0.2007 
## F-statistic: 5.645 on 6 and 105 DF,  p-value: 4.112e-05</code></pre>
<p>We see from the results of this linear model that there is very moderate evidence of an interaction effect between BMI group and nationality such that those with African nationality have a greater difference in diversity between the lean and overweight groups than those of African American nationality. Overall, the variables in the model accounted for 24.4% of variability in the diversity response.</p>
</div>
<div id="model-diagnostics" class="section level4 hasAnchor">
<h4>Model Diagnostics<a href="module-3-modeling-part-1.html#model-diagnostics" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Again, after fitting the model, we must check for the following assumptions:</p>
<ol style="list-style-type: decimal">
<li><p>Linearity - we already checked this prior to fitting the model</p></li>
<li><p>Independence - we already fixed this by removing the second time point</p></li>
<li><p>Homoscedasticity - we can check this with a scatterplot of model residuals vs. fitted values</p></li>
<li><p>Normality - we can check this with a qq plot of model residuals</p></li>
<li><p>No multicollinearity - we will ignore this for this model for now since we have interaction terms and look at it for the model without interactions</p></li>
</ol>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="module-3-modeling-part-1.html#cb139-1" tabindex="-1"></a><span class="co"># 3. homoscedasticity</span></span>
<span id="cb139-2"><a href="module-3-modeling-part-1.html#cb139-2" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">residuals</span>(mod1)</span>
<span id="cb139-3"><a href="module-3-modeling-part-1.html#cb139-3" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">fitted</span>(mod1)</span>
<span id="cb139-4"><a href="module-3-modeling-part-1.html#cb139-4" tabindex="-1"></a>mod_res <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(res,fit)</span>
<span id="cb139-5"><a href="module-3-modeling-part-1.html#cb139-5" tabindex="-1"></a><span class="co"># create the residuals vs fitted values plot</span></span>
<span id="cb139-6"><a href="module-3-modeling-part-1.html#cb139-6" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>mod_res,<span class="fu">aes</span>(<span class="at">x =</span> fit, <span class="at">y =</span> res)) <span class="sc">+</span></span>
<span id="cb139-7"><a href="module-3-modeling-part-1.html#cb139-7" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span>  <span class="co"># Plot the residuals</span></span>
<span id="cb139-8"><a href="module-3-modeling-part-1.html#cb139-8" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;red&quot;</span>) <span class="sc">+</span>  <span class="co"># Add a horizontal line at 0</span></span>
<span id="cb139-9"><a href="module-3-modeling-part-1.html#cb139-9" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Residuals vs Fitted Values&quot;</span>,</span>
<span id="cb139-10"><a href="module-3-modeling-part-1.html#cb139-10" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Fitted Values&quot;</span>,</span>
<span id="cb139-11"><a href="module-3-modeling-part-1.html#cb139-11" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Residuals&quot;</span>) <span class="sc">+</span></span>
<span id="cb139-12"><a href="module-3-modeling-part-1.html#cb139-12" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="module-3-modeling-part-1.html#cb140-1" tabindex="-1"></a><span class="co"># 4. normality </span></span>
<span id="cb140-2"><a href="module-3-modeling-part-1.html#cb140-2" tabindex="-1"></a><span class="co"># QQ-plot to assess normality</span></span>
<span id="cb140-3"><a href="module-3-modeling-part-1.html#cb140-3" tabindex="-1"></a><span class="fu">qqPlot</span>(mod_res<span class="sc">$</span>res)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-22-2.png" width="672" /></p>
<pre><code>## [1] 30 97</code></pre>
</div>
<div id="comparing-nested-models" class="section level4 hasAnchor">
<h4>Comparing Nested Models<a href="module-3-modeling-part-1.html#comparing-nested-models" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>When fitting multiple linear regression models, it’s often useful to explore whether a simpler, more parsimonious model can adequately explain the data. For example, when working with indicator variables, we can assess whether they should be included for a change in intercept (main effect), a change in slope (interaction effect), or excluded altogether. By comparing models without the indicator, with only the main effect, and with the interaction effect, we can evaluate the contribution of the indicator variable and determine its necessity in the final model.</p>
<p>Here, we will use the anova() function in R to compare nested models. The null hypothesis of the test run by this anova function is H0: reduced (simpler model) is adequate and the alternative hypothesis is H1: Full (more complex) model is needed to explain the data. Therefore, if the p value is small, then we have evidence that we should include the extra variable(s).</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="module-3-modeling-part-1.html#cb142-1" tabindex="-1"></a><span class="co"># fit model with only main effects for nationality</span></span>
<span id="cb142-2"><a href="module-3-modeling-part-1.html#cb142-2" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Shannon<span class="sc">~</span>nationality<span class="sc">+</span>bmi_group<span class="sc">+</span>sex, <span class="at">data=</span>dietswap_data_t1)</span>
<span id="cb142-3"><a href="module-3-modeling-part-1.html#cb142-3" tabindex="-1"></a><span class="fu">summary</span>(mod2)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = Shannon ~ nationality + bmi_group + sex, data = dietswap_data_t1)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.35941 -0.34599  0.06785  0.32141  1.51472 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)          2.97433    0.19853  14.982  &lt; 2e-16 ***
## nationalityAFR      -0.52438    0.11508  -4.557 1.39e-05 ***
## bmi_groupoverweight -0.04098    0.16653  -0.246   0.8061    
## bmi_groupobese      -0.31923    0.16734  -1.908   0.0591 .  
## sexmale             -0.12343    0.13122  -0.941   0.3490    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.5606 on 107 degrees of freedom
## Multiple R-squared:  0.2029, Adjusted R-squared:  0.1731 
## F-statistic:  6.81 on 4 and 107 DF,  p-value: 6.374e-05</code></pre>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="module-3-modeling-part-1.html#cb144-1" tabindex="-1"></a><span class="co"># fit model without nationality</span></span>
<span id="cb144-2"><a href="module-3-modeling-part-1.html#cb144-2" tabindex="-1"></a>mod3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Shannon<span class="sc">~</span>bmi_group<span class="sc">+</span>sex, <span class="at">data=</span>dietswap_data_t1)</span>
<span id="cb144-3"><a href="module-3-modeling-part-1.html#cb144-3" tabindex="-1"></a><span class="fu">summary</span>(mod3)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = Shannon ~ bmi_group + sex, data = dietswap_data_t1)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -1.4473 -0.4295  0.1259  0.4826  1.3520 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)          2.46753    0.17886  13.796   &lt;2e-16 ***
## bmi_groupoverweight  0.23528    0.16870   1.395    0.166    
## bmi_groupobese      -0.07057    0.17206  -0.410    0.682    
## sexmale              0.02172    0.13846   0.157    0.876    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.6098 on 108 degrees of freedom
## Multiple R-squared:  0.04824,    Adjusted R-squared:  0.02181 
## F-statistic: 1.825 on 3 and 108 DF,  p-value: 0.147</code></pre>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="module-3-modeling-part-1.html#cb146-1" tabindex="-1"></a><span class="co"># compare nested models</span></span>
<span id="cb146-2"><a href="module-3-modeling-part-1.html#cb146-2" tabindex="-1"></a><span class="fu">anova</span>(mod3,mod2)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Model 1: Shannon ~ bmi_group + sex
## Model 2: Shannon ~ nationality + bmi_group + sex
##   Res.Df    RSS Df Sum of Sq      F    Pr(&gt;F)    
## 1    108 40.155                                  
## 2    107 33.629  1    6.5256 20.763 1.385e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="module-3-modeling-part-1.html#cb148-1" tabindex="-1"></a><span class="fu">anova</span>(mod3,mod1)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Model 1: Shannon ~ bmi_group + sex
## Model 2: Shannon ~ nationality * bmi_group + sex
##   Res.Df    RSS Df Sum of Sq      F    Pr(&gt;F)    
## 1    108 40.155                                  
## 2    105 31.901  3    8.2543 9.0563 2.189e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="module-3-modeling-part-1.html#cb150-1" tabindex="-1"></a><span class="fu">anova</span>(mod2,mod1)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Model 1: Shannon ~ nationality + bmi_group + sex
## Model 2: Shannon ~ nationality * bmi_group + sex
##   Res.Df    RSS Df Sum of Sq      F  Pr(&gt;F)  
## 1    107 33.629                              
## 2    105 31.901  2    1.7287 2.8449 0.06263 .
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>From comparing the nested models, we see that nationality should be included in the model but that we do not require the interaction effect. We can go ahead and check the model diagnostics for mod2.</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="module-3-modeling-part-1.html#cb152-1" tabindex="-1"></a><span class="co"># 3. homoscedasticity</span></span>
<span id="cb152-2"><a href="module-3-modeling-part-1.html#cb152-2" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">residuals</span>(mod2)</span>
<span id="cb152-3"><a href="module-3-modeling-part-1.html#cb152-3" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">fitted</span>(mod2)</span>
<span id="cb152-4"><a href="module-3-modeling-part-1.html#cb152-4" tabindex="-1"></a>mod_res <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(res,fit)</span>
<span id="cb152-5"><a href="module-3-modeling-part-1.html#cb152-5" tabindex="-1"></a><span class="co"># create the residuals vs fitted values plot</span></span>
<span id="cb152-6"><a href="module-3-modeling-part-1.html#cb152-6" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>mod_res,<span class="fu">aes</span>(<span class="at">x =</span> fit, <span class="at">y =</span> res)) <span class="sc">+</span></span>
<span id="cb152-7"><a href="module-3-modeling-part-1.html#cb152-7" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span>  <span class="co"># Plot the residuals</span></span>
<span id="cb152-8"><a href="module-3-modeling-part-1.html#cb152-8" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;red&quot;</span>) <span class="sc">+</span>  <span class="co"># Add a horizontal line at 0</span></span>
<span id="cb152-9"><a href="module-3-modeling-part-1.html#cb152-9" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Residuals vs Fitted Values&quot;</span>,</span>
<span id="cb152-10"><a href="module-3-modeling-part-1.html#cb152-10" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Fitted Values&quot;</span>,</span>
<span id="cb152-11"><a href="module-3-modeling-part-1.html#cb152-11" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Residuals&quot;</span>) <span class="sc">+</span></span>
<span id="cb152-12"><a href="module-3-modeling-part-1.html#cb152-12" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="module-3-modeling-part-1.html#cb153-1" tabindex="-1"></a><span class="co"># 4.normality </span></span>
<span id="cb153-2"><a href="module-3-modeling-part-1.html#cb153-2" tabindex="-1"></a><span class="co"># QQ-plot to assess normality</span></span>
<span id="cb153-3"><a href="module-3-modeling-part-1.html#cb153-3" tabindex="-1"></a><span class="fu">qqPlot</span>(mod_res<span class="sc">$</span>res)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-2.png" width="672" /></p>
<pre><code>## [1] 30 97</code></pre>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="module-3-modeling-part-1.html#cb155-1" tabindex="-1"></a><span class="co"># 5. no multicollinearity</span></span>
<span id="cb155-2"><a href="module-3-modeling-part-1.html#cb155-2" tabindex="-1"></a><span class="fu">vif</span>(mod2)</span></code></pre></div>
<pre><code>##                 GVIF Df GVIF^(1/(2*Df))
## nationality 1.166324  1        1.079965
## bmi_group   1.667090  2        1.136291
## sex         1.521859  1        1.233637</code></pre>
</div>
<div id="visualize-our-model" class="section level4 hasAnchor">
<h4>Visualize our Model<a href="module-3-modeling-part-1.html#visualize-our-model" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now let’s plot the interaction between bmi_group and nationality on Shannon diversity to help us interpret our findings.</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="module-3-modeling-part-1.html#cb157-1" tabindex="-1"></a><span class="fu">ggplot</span>(dietswap_data_t1, <span class="fu">aes</span>(<span class="at">x =</span> bmi_group, <span class="at">y =</span> Shannon, <span class="at">color =</span> nationality, <span class="at">group =</span> nationality)) <span class="sc">+</span></span>
<span id="cb157-2"><a href="module-3-modeling-part-1.html#cb157-2" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> mean, <span class="at">geom =</span> <span class="st">&quot;point&quot;</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Plot the means</span></span>
<span id="cb157-3"><a href="module-3-modeling-part-1.html#cb157-3" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun =</span> mean, <span class="at">geom =</span> <span class="st">&quot;line&quot;</span>) <span class="sc">+</span>             <span class="co"># Connect the means with lines</span></span>
<span id="cb157-4"><a href="module-3-modeling-part-1.html#cb157-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Interaction between Nationality, BMI Group, and Shannon Diversity&quot;</span>,</span>
<span id="cb157-5"><a href="module-3-modeling-part-1.html#cb157-5" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;BMI Group&quot;</span>,</span>
<span id="cb157-6"><a href="module-3-modeling-part-1.html#cb157-6" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Shannon Diversity Index&quot;</span>,</span>
<span id="cb157-7"><a href="module-3-modeling-part-1.html#cb157-7" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&quot;Nationality&quot;</span>) <span class="sc">+</span></span>
<span id="cb157-8"><a href="module-3-modeling-part-1.html#cb157-8" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
</div>
</div>
<div id="logistic-regression-1" class="section level3 hasAnchor">
<h3>Logistic Regression<a href="module-3-modeling-part-1.html#logistic-regression-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Suppose researchers are interested to learn more about the association between shannon diversity of the throat bacteria and smoking. We can use logistic regression to help answer this question through looking at whether we can predict smoking status with shannon diversity of bacteria in the throat, age, and sex.</p>
<div id="loading-the-data-1" class="section level4 hasAnchor">
<h4>Loading the Data<a href="module-3-modeling-part-1.html#loading-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="module-3-modeling-part-1.html#cb158-1" tabindex="-1"></a><span class="co"># first, need to load in meta data and otu table</span></span>
<span id="cb158-2"><a href="module-3-modeling-part-1.html#cb158-2" tabindex="-1"></a><span class="fu">data</span>(throat.meta)</span>
<span id="cb158-3"><a href="module-3-modeling-part-1.html#cb158-3" tabindex="-1"></a><span class="fu">data</span>(throat.otu.tab)</span>
<span id="cb158-4"><a href="module-3-modeling-part-1.html#cb158-4" tabindex="-1"></a></span>
<span id="cb158-5"><a href="module-3-modeling-part-1.html#cb158-5" tabindex="-1"></a><span class="co"># look up descriptions of the data</span></span>
<span id="cb158-6"><a href="module-3-modeling-part-1.html#cb158-6" tabindex="-1"></a>?throat.meta</span>
<span id="cb158-7"><a href="module-3-modeling-part-1.html#cb158-7" tabindex="-1"></a>?throat.otu.tab</span>
<span id="cb158-8"><a href="module-3-modeling-part-1.html#cb158-8" tabindex="-1"></a></span>
<span id="cb158-9"><a href="module-3-modeling-part-1.html#cb158-9" tabindex="-1"></a><span class="co"># let&#39;s turn this data into a phyloseq object</span></span>
<span id="cb158-10"><a href="module-3-modeling-part-1.html#cb158-10" tabindex="-1"></a>throat.physeq <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(</span>
<span id="cb158-11"><a href="module-3-modeling-part-1.html#cb158-11" tabindex="-1"></a>  <span class="fu">otu_table</span>(throat.otu.tab, <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>),</span>
<span id="cb158-12"><a href="module-3-modeling-part-1.html#cb158-12" tabindex="-1"></a>  <span class="fu">sample_data</span>(throat.meta)</span>
<span id="cb158-13"><a href="module-3-modeling-part-1.html#cb158-13" tabindex="-1"></a>)</span>
<span id="cb158-14"><a href="module-3-modeling-part-1.html#cb158-14" tabindex="-1"></a></span>
<span id="cb158-15"><a href="module-3-modeling-part-1.html#cb158-15" tabindex="-1"></a><span class="co"># check our phyloseq object</span></span>
<span id="cb158-16"><a href="module-3-modeling-part-1.html#cb158-16" tabindex="-1"></a><span class="fu">otu_table</span>(throat.physeq)</span>
<span id="cb158-17"><a href="module-3-modeling-part-1.html#cb158-17" tabindex="-1"></a><span class="fu">sample_data</span>(throat.physeq)</span>
<span id="cb158-18"><a href="module-3-modeling-part-1.html#cb158-18" tabindex="-1"></a></span>
<span id="cb158-19"><a href="module-3-modeling-part-1.html#cb158-19" tabindex="-1"></a></span>
<span id="cb158-20"><a href="module-3-modeling-part-1.html#cb158-20" tabindex="-1"></a><span class="co"># now we can estimate shannon diversity</span></span>
<span id="cb158-21"><a href="module-3-modeling-part-1.html#cb158-21" tabindex="-1"></a>Shannon <span class="ot">&lt;-</span> <span class="fu">estimate_richness</span>(<span class="fu">otu_table</span>(throat.physeq), <span class="at">measures=</span><span class="st">&quot;Shannon&quot;</span>)</span>
<span id="cb158-22"><a href="module-3-modeling-part-1.html#cb158-22" tabindex="-1"></a>throat_data <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(<span class="at">Shannon=</span>Shannon<span class="sc">$</span>Shannon,<span class="fu">sample_data</span>(throat.physeq))</span></code></pre></div>
</div>
<div id="fitting-the-logistic-regression-model" class="section level4 hasAnchor">
<h4>Fitting the Logistic Regression Model<a href="module-3-modeling-part-1.html#fitting-the-logistic-regression-model" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In R, we can use the glm() function to fit a logistic regression model by setting family = “binomial” as an argument.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="module-3-modeling-part-1.html#cb159-1" tabindex="-1"></a><span class="co"># now let&#39;s look at the effect of sex, age, and diversity on smoking</span></span>
<span id="cb159-2"><a href="module-3-modeling-part-1.html#cb159-2" tabindex="-1"></a>log_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(SmokingStatus<span class="sc">~</span>Age<span class="sc">+</span>Sex<span class="sc">+</span>Shannon, <span class="at">data=</span>throat_data, <span class="at">family=</span>binomial)</span>
<span id="cb159-3"><a href="module-3-modeling-part-1.html#cb159-3" tabindex="-1"></a><span class="fu">summary</span>(log_mod)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = SmokingStatus ~ Age + Sex + Shannon, family = binomial, 
##     data = throat_data)
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)  
## (Intercept)  1.97623    2.54902   0.775   0.4382  
## Age          0.04757    0.02798   1.700   0.0891 .
## SexMale      0.80698    0.61972   1.302   0.1929  
## Shannon     -1.29961    0.80217  -1.620   0.1052  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 82.911  on 59  degrees of freedom
## Residual deviance: 75.147  on 56  degrees of freedom
## AIC: 83.147
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="module-3-modeling-part-1.html#cb161-1" tabindex="-1"></a><span class="co"># get odds ratios</span></span>
<span id="cb161-2"><a href="module-3-modeling-part-1.html#cb161-2" tabindex="-1"></a><span class="fu">exp</span>(<span class="fu">coef</span>(log_mod))</span></code></pre></div>
<pre><code>## (Intercept)         Age     SexMale     Shannon 
##   7.2154548   1.0487182   2.2411336   0.2726372</code></pre>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="module-3-modeling-part-1.html#cb163-1" tabindex="-1"></a><span class="co"># 95% confidence intervals for odds ratios</span></span>
<span id="cb163-2"><a href="module-3-modeling-part-1.html#cb163-2" tabindex="-1"></a><span class="fu">exp</span>(<span class="fu">confint</span>(log_mod))</span></code></pre></div>
<pre><code>##                  2.5 %      97.5 %
## (Intercept) 0.05231538 1437.791848
## Age         0.99472215    1.111987
## SexMale     0.68252995    7.976087
## Shannon     0.05015106    1.235533</code></pre>
<p>It does not look like Shannon diversity has a significant association with smoking status. That is, the p-value for the Shannon diversity covariate is 0.105. We can interpret the coefficient on the logit scale as, increasing Shannon diversity by 1 unit is estimated to decrease log(p/1-p) of smoking by 1.2996. Alternatively, we can interpret the coefficient on the odds scale by exponentiating the coefficient. On the odds scale, we can say that increasing the Shannon diversity changes the odds of smoking by a factor of 0.273 (95% CI: 0.050, 1.236).</p>
</div>
<div id="model-diagnostic" class="section level4 hasAnchor">
<h4>Model Diagnostic<a href="module-3-modeling-part-1.html#model-diagnostic" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Again, after fitting the model, we must check for the following assumptions:</p>
<ol style="list-style-type: decimal">
<li><p>Linearity - we can check the relationship between covariates and logit outcome with component-residual plots</p></li>
<li><p>Independence - we will assume the data observations are independent</p></li>
<li><p>Proper fitting distribution - we can check this with a scatterplot of model DEVIANCE residuals vs. linear predictor</p></li>
<li><p>No multicollinearity - we can check this with variance inflation factors</p></li>
</ol>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="module-3-modeling-part-1.html#cb165-1" tabindex="-1"></a><span class="co"># checking assumptions</span></span>
<span id="cb165-2"><a href="module-3-modeling-part-1.html#cb165-2" tabindex="-1"></a><span class="co"># 1. linearity assumption with component+residual plots</span></span>
<span id="cb165-3"><a href="module-3-modeling-part-1.html#cb165-3" tabindex="-1"></a><span class="fu">crPlot</span>(log_mod, <span class="at">variable=</span><span class="st">&quot;Shannon&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="module-3-modeling-part-1.html#cb166-1" tabindex="-1"></a><span class="fu">crPlot</span>(log_mod, <span class="at">variable=</span><span class="st">&quot;Age&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-28-2.png" width="672" /></p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="module-3-modeling-part-1.html#cb167-1" tabindex="-1"></a><span class="co"># 3. proper fitting distribution (look for no pattern/constant variance)</span></span>
<span id="cb167-2"><a href="module-3-modeling-part-1.html#cb167-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">residuals</span>(log_mod)<span class="sc">~</span><span class="fu">predict</span>(log_mod)) <span class="co"># notice it looks weird because we only have 2 response options</span></span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-28-3.png" width="672" /></p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="module-3-modeling-part-1.html#cb168-1" tabindex="-1"></a><span class="fu">binnedplot</span>(<span class="fu">predict</span>(log_mod), <span class="fu">residuals</span>(log_mod))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-28-4.png" width="672" /></p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="module-3-modeling-part-1.html#cb169-1" tabindex="-1"></a><span class="co"># 4. no multicollinearity</span></span>
<span id="cb169-2"><a href="module-3-modeling-part-1.html#cb169-2" tabindex="-1"></a><span class="fu">vif</span>(log_mod) <span class="co"># a bit concerning - could be too high of a correlation</span></span></code></pre></div>
<pre><code>##      Age      Sex  Shannon 
## 1.074958 1.120760 1.095087</code></pre>
</div>
<div id="visualization" class="section level4 hasAnchor">
<h4>Visualization<a href="module-3-modeling-part-1.html#visualization" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="module-3-modeling-part-1.html#cb171-1" tabindex="-1"></a><span class="co"># first we need smoking status to be numeric</span></span>
<span id="cb171-2"><a href="module-3-modeling-part-1.html#cb171-2" tabindex="-1"></a>throat_data<span class="sc">$</span>SmokingStatusNumeric <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(throat_data<span class="sc">$</span>SmokingStatus<span class="sc">==</span><span class="st">&quot;Smoker&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb171-3"><a href="module-3-modeling-part-1.html#cb171-3" tabindex="-1"></a><span class="co"># create a ggplot object with the data points</span></span>
<span id="cb171-4"><a href="module-3-modeling-part-1.html#cb171-4" tabindex="-1"></a></span>
<span id="cb171-5"><a href="module-3-modeling-part-1.html#cb171-5" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(throat_data, <span class="fu">aes</span>(<span class="at">x =</span> Shannon, <span class="at">y =</span> SmokingStatusNumeric)) <span class="sc">+</span></span>
<span id="cb171-6"><a href="module-3-modeling-part-1.html#cb171-6" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span>  <span class="co"># Plot the actual data points</span></span>
<span id="cb171-7"><a href="module-3-modeling-part-1.html#cb171-7" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">&quot;glm&quot;</span>, <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family =</span> binomial), <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">color =</span> <span class="st">&quot;blue&quot;</span>) <span class="sc">+</span></span>
<span id="cb171-8"><a href="module-3-modeling-part-1.html#cb171-8" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Logistic Regression Curve&quot;</span>,</span>
<span id="cb171-9"><a href="module-3-modeling-part-1.html#cb171-9" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Shannon Diversity&quot;</span>,</span>
<span id="cb171-10"><a href="module-3-modeling-part-1.html#cb171-10" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Probability of Smoking&quot;</span>)</span>
<span id="cb171-11"><a href="module-3-modeling-part-1.html#cb171-11" tabindex="-1"></a></span>
<span id="cb171-12"><a href="module-3-modeling-part-1.html#cb171-12" tabindex="-1"></a><span class="co"># display the plot</span></span>
<span id="cb171-13"><a href="module-3-modeling-part-1.html#cb171-13" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
</div>
</div>
<div id="binomial-regression" class="section level3 hasAnchor">
<h3>Binomial Regression<a href="module-3-modeling-part-1.html#binomial-regression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Suppose researchers are interested in the number of surviving trout eggs in boxes across different locations at different weeks after placement. In particular, the outcome of interest is the number of trout eggs alive in the box/the total number of trout eggs in the box.</p>
<div id="loading-the-data-2" class="section level4 hasAnchor">
<h4>Loading the Data<a href="module-3-modeling-part-1.html#loading-the-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We will use the troutegg data from faraway for this analysis.</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="module-3-modeling-part-1.html#cb172-1" tabindex="-1"></a><span class="co"># first, let&#39;s load in the data</span></span>
<span id="cb172-2"><a href="module-3-modeling-part-1.html#cb172-2" tabindex="-1"></a><span class="fu">data</span>(troutegg) <span class="co"># from faraway</span></span>
<span id="cb172-3"><a href="module-3-modeling-part-1.html#cb172-3" tabindex="-1"></a></span>
<span id="cb172-4"><a href="module-3-modeling-part-1.html#cb172-4" tabindex="-1"></a><span class="co"># view description of the data</span></span>
<span id="cb172-5"><a href="module-3-modeling-part-1.html#cb172-5" tabindex="-1"></a>?troutegg</span>
<span id="cb172-6"><a href="module-3-modeling-part-1.html#cb172-6" tabindex="-1"></a></span>
<span id="cb172-7"><a href="module-3-modeling-part-1.html#cb172-7" tabindex="-1"></a><span class="co"># view data</span></span>
<span id="cb172-8"><a href="module-3-modeling-part-1.html#cb172-8" tabindex="-1"></a><span class="fu">head</span>(troutegg)</span></code></pre></div>
<pre><code>##   survive total location period
## 1      89    94        1      4
## 2     106   108        2      4
## 3     119   123        3      4
## 4     104   104        4      4
## 5      49    93        5      4
## 6      94    98        1      7</code></pre>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="module-3-modeling-part-1.html#cb174-1" tabindex="-1"></a><span class="fu">str</span>(troutegg)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    20 obs. of  4 variables:
##  $ survive : int  89 106 119 104 49 94 91 100 80 11 ...
##  $ total   : int  94 108 123 104 93 98 106 130 97 113 ...
##  $ location: Factor w/ 5 levels &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,..: 1 2 3 4 5 1 2 3 4 5 ...
##  $ period  : Factor w/ 4 levels &quot;4&quot;,&quot;7&quot;,&quot;8&quot;,&quot;11&quot;: 1 1 1 1 1 2 2 2 2 2 ...</code></pre>
</div>
<div id="fitting-the-binomial-regression" class="section level4 hasAnchor">
<h4>Fitting the binomial regression<a href="module-3-modeling-part-1.html#fitting-the-binomial-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We will fit the binomial regression using the glm function and setting family=“binomial”. However, we have to tell R how many successes we have in our outcome and how many failures.</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="module-3-modeling-part-1.html#cb176-1" tabindex="-1"></a><span class="co"># fit binomial</span></span>
<span id="cb176-2"><a href="module-3-modeling-part-1.html#cb176-2" tabindex="-1"></a>binom_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(survive,total<span class="sc">-</span>survive)<span class="sc">~</span>location<span class="sc">+</span>period, <span class="at">data=</span>troutegg, <span class="at">family=</span><span class="st">&quot;binomial&quot;</span>)</span>
<span id="cb176-3"><a href="module-3-modeling-part-1.html#cb176-3" tabindex="-1"></a><span class="fu">summary</span>(binom_mod)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = cbind(survive, total - survive) ~ location + period, 
##     family = &quot;binomial&quot;, data = troutegg)
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)   4.6358     0.2813  16.479  &lt; 2e-16 ***
## location2    -0.4168     0.2461  -1.694   0.0903 .  
## location3    -1.2421     0.2194  -5.660 1.51e-08 ***
## location4    -0.9509     0.2288  -4.157 3.23e-05 ***
## location5    -4.6138     0.2502 -18.439  &lt; 2e-16 ***
## period7      -2.1702     0.2384  -9.103  &lt; 2e-16 ***
## period8      -2.3256     0.2429  -9.573  &lt; 2e-16 ***
## period11     -2.4500     0.2341 -10.466  &lt; 2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 1021.469  on 19  degrees of freedom
## Residual deviance:   64.495  on 12  degrees of freedom
## AIC: 157.03
## 
## Number of Fisher Scoring iterations: 5</code></pre>
</div>
<div id="model-diagnostics-1" class="section level4 hasAnchor">
<h4>Model diagnostics<a href="module-3-modeling-part-1.html#model-diagnostics-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Again, after fitting the model, we must check for the following assumptions:</p>
<ol style="list-style-type: decimal">
<li><p>Linearity - we do not have any continuous or numeric variables</p></li>
<li><p>Independence - we will assume the data observations are independent</p></li>
<li><p>Proper fitting distribution - we can check this with a scatterplot of model DEVIANCE residuals vs. linear predictor</p></li>
<li><p>No multicollinearity - we can check this with variance inflation factors</p></li>
</ol>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="module-3-modeling-part-1.html#cb178-1" tabindex="-1"></a><span class="co"># checking assumptions</span></span>
<span id="cb178-2"><a href="module-3-modeling-part-1.html#cb178-2" tabindex="-1"></a><span class="co"># 3. proper fitting distribution (look for no pattern/constant variance)</span></span>
<span id="cb178-3"><a href="module-3-modeling-part-1.html#cb178-3" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">residuals</span>(binom_mod)<span class="sc">~</span><span class="fu">predict</span>(binom_mod)) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="module-3-modeling-part-1.html#cb179-1" tabindex="-1"></a><span class="co"># 4. No multicollinearity</span></span>
<span id="cb179-2"><a href="module-3-modeling-part-1.html#cb179-2" tabindex="-1"></a><span class="fu">vif</span>(binom_mod) </span></code></pre></div>
<pre><code>##             GVIF Df GVIF^(1/(2*Df))
## location 1.33796  4        1.037064
## period   1.33796  3        1.049721</code></pre>
</div>
<div id="addressing-overdispersion-in-binomial" class="section level4 hasAnchor">
<h4>Addressing Overdispersion in Binomial<a href="module-3-modeling-part-1.html#addressing-overdispersion-in-binomial" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The binomial model assumes that the variance is related to the our outcome, p, as p(1-p). If the variance is greater than what is assumed by the model, it is called ‘overdispersion’. We can estimate overdispersion as deviance/df residuals. If this ratio is greater than 1 we have overdispersion, if it is (roughly) 1 we have no overdispersion, and if it is less than 1, we have underdispersion.</p>
<p>If we have overdispersion of the binomial, we can try to fit a quasibinomial model instead which will estimate the dispersion parameter instead of assuming it to be 1.</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="module-3-modeling-part-1.html#cb181-1" tabindex="-1"></a><span class="co"># estimating the dispersion</span></span>
<span id="cb181-2"><a href="module-3-modeling-part-1.html#cb181-2" tabindex="-1"></a>binom_mod<span class="sc">$</span>deviance<span class="sc">/</span>binom_mod<span class="sc">$</span>df.residual</span></code></pre></div>
<pre><code>## [1] 5.374593</code></pre>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="module-3-modeling-part-1.html#cb183-1" tabindex="-1"></a><span class="co"># if it is large, we can use &#39;quasibinomial&#39; as the family instead to account for this</span></span>
<span id="cb183-2"><a href="module-3-modeling-part-1.html#cb183-2" tabindex="-1"></a>binom_mod2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(survive,total<span class="sc">-</span>survive)<span class="sc">~</span>location<span class="sc">+</span>period, <span class="at">data=</span>troutegg, <span class="at">family=</span><span class="st">&quot;quasibinomial&quot;</span>)</span>
<span id="cb183-3"><a href="module-3-modeling-part-1.html#cb183-3" tabindex="-1"></a><span class="fu">summary</span>(binom_mod2)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = cbind(survive, total - survive) ~ location + period, 
##     family = &quot;quasibinomial&quot;, data = troutegg)
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)   4.6358     0.6495   7.138 1.18e-05 ***
## location2    -0.4168     0.5682  -0.734 0.477315    
## location3    -1.2421     0.5066  -2.452 0.030501 *  
## location4    -0.9509     0.5281  -1.800 0.096970 .  
## location5    -4.6138     0.5777  -7.987 3.82e-06 ***
## period7      -2.1702     0.5504  -3.943 0.001953 ** 
## period8      -2.3256     0.5609  -4.146 0.001356 ** 
## period11     -2.4500     0.5405  -4.533 0.000686 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasibinomial family taken to be 5.330358)
## 
##     Null deviance: 1021.469  on 19  degrees of freedom
## Residual deviance:   64.495  on 12  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 5</code></pre>
</div>
</div>
<div id="poisson-regression" class="section level3 hasAnchor">
<h3>Poisson Regression<a href="module-3-modeling-part-1.html#poisson-regression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="loading-the-data-3" class="section level4 hasAnchor">
<h4>Loading the Data<a href="module-3-modeling-part-1.html#loading-the-data-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We will use the faramea.csv data.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="module-3-modeling-part-1.html#cb185-1" tabindex="-1"></a><span class="co"># read in the dataset</span></span>
<span id="cb185-2"><a href="module-3-modeling-part-1.html#cb185-2" tabindex="-1"></a>faramea <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;module3_faramea.csv&#39;</span>)</span>
<span id="cb185-3"><a href="module-3-modeling-part-1.html#cb185-3" tabindex="-1"></a></span>
<span id="cb185-4"><a href="module-3-modeling-part-1.html#cb185-4" tabindex="-1"></a><span class="co"># look at first 6 rows</span></span>
<span id="cb185-5"><a href="module-3-modeling-part-1.html#cb185-5" tabindex="-1"></a><span class="fu">head</span>(faramea)</span></code></pre></div>
<pre><code>##     X   UTM.EW  UTM.NS Precipitation Elevation Age Age.cat Geology
## 1  B0 625754.0 1011569        2530.0       120   3      c3      Tb
## 2 B49 626654.0 1011969        2530.0       120   3      c3      Tb
## 3  p1 614856.9 1031786        2993.2        20   2      c2      Tc
## 4  p2 613985.4 1030725        3072.0       100   3      c3      Tc
## 5  p3 614674.3 1023802        3007.4       180   1      c1      Tc
## 6  p4 615018.6 1023548        2999.8       180   1      c1      Tc
##   Faramea.occidentalis
## 1                   14
## 2                    7
## 3                    0
## 4                    0
## 5                    2
## 6                    1</code></pre>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="module-3-modeling-part-1.html#cb187-1" tabindex="-1"></a><span class="co"># look at structure</span></span>
<span id="cb187-2"><a href="module-3-modeling-part-1.html#cb187-2" tabindex="-1"></a><span class="fu">str</span>(faramea)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    43 obs. of  9 variables:
##  $ X                   : chr  &quot;B0&quot; &quot;B49&quot; &quot;p1&quot; &quot;p2&quot; ...
##  $ UTM.EW              : num  625754 626654 614857 613985 614674 ...
##  $ UTM.NS              : num  1011569 1011969 1031786 1030725 1023802 ...
##  $ Precipitation       : num  2530 2530 2993 3072 3007 ...
##  $ Elevation           : int  120 120 20 100 180 180 40 30 60 50 ...
##  $ Age                 : int  3 3 2 3 1 1 2 2 1 3 ...
##  $ Age.cat             : chr  &quot;c3&quot; &quot;c3&quot; &quot;c2&quot; &quot;c3&quot; ...
##  $ Geology             : chr  &quot;Tb&quot; &quot;Tb&quot; &quot;Tc&quot; &quot;Tc&quot; ...
##  $ Faramea.occidentalis: int  14 7 0 0 2 1 0 0 2 0 ...</code></pre>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="module-3-modeling-part-1.html#cb189-1" tabindex="-1"></a><span class="co"># look at summary of variables</span></span>
<span id="cb189-2"><a href="module-3-modeling-part-1.html#cb189-2" tabindex="-1"></a><span class="fu">summary</span>(faramea)</span></code></pre></div>
<pre><code>##       X                 UTM.EW           UTM.NS        Precipitation 
##  Length:43          Min.   :600714   Min.   : 962862   Min.   :1888  
##  Class :character   1st Qu.:625578   1st Qu.:1009776   1st Qu.:2406  
##  Mode  :character   Median :637732   Median :1012428   Median :2516  
##                     Mean   :635927   Mean   :1013984   Mean   :2643  
##                     3rd Qu.:645611   3rd Qu.:1023675   3rd Qu.:2996  
##                     Max.   :688165   Max.   :1045987   Max.   :4002  
##    Elevation          Age          Age.cat            Geology         
##  Min.   : 10.0   Min.   :1.000   Length:43          Length:43         
##  1st Qu.: 59.0   1st Qu.:1.000   Class :character   Class :character  
##  Median :120.0   Median :2.000   Mode  :character   Mode  :character  
##  Mean   :192.3   Mean   :2.093                                        
##  3rd Qu.:180.0   3rd Qu.:3.000                                        
##  Max.   :830.0   Max.   :3.000                                        
##  Faramea.occidentalis
##  Min.   : 0.000      
##  1st Qu.: 0.000      
##  Median : 0.000      
##  Mean   : 3.884      
##  3rd Qu.: 4.500      
##  Max.   :42.000</code></pre>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="module-3-modeling-part-1.html#cb191-1" tabindex="-1"></a><span class="co"># convert age to numeric (helps with crPlots)</span></span>
<span id="cb191-2"><a href="module-3-modeling-part-1.html#cb191-2" tabindex="-1"></a>faramea<span class="sc">$</span>Age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(faramea<span class="sc">$</span>Age)</span></code></pre></div>
</div>
<div id="fitting-the-poisson-regression" class="section level4 hasAnchor">
<h4>Fitting the Poisson Regression<a href="module-3-modeling-part-1.html#fitting-the-poisson-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Again, we will be using the glm() function here to fit our Poisson regression model and this time, we will set family=“poisson”.</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="module-3-modeling-part-1.html#cb192-1" tabindex="-1"></a><span class="co"># fitting the Poisson model</span></span>
<span id="cb192-2"><a href="module-3-modeling-part-1.html#cb192-2" tabindex="-1"></a>glm.poisson <span class="ot">=</span> <span class="fu">glm</span>(Faramea.occidentalis <span class="sc">~</span> Elevation<span class="sc">+</span>Age<span class="sc">+</span>Precipitation,</span>
<span id="cb192-3"><a href="module-3-modeling-part-1.html#cb192-3" tabindex="-1"></a>                  <span class="at">data =</span> faramea,</span>
<span id="cb192-4"><a href="module-3-modeling-part-1.html#cb192-4" tabindex="-1"></a>                  <span class="at">family =</span> poisson)</span>
<span id="cb192-5"><a href="module-3-modeling-part-1.html#cb192-5" tabindex="-1"></a><span class="fu">summary</span>(glm.poisson)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = Faramea.occidentalis ~ Elevation + Age + Precipitation, 
##     family = poisson, data = faramea)
## 
## Coefficients:
##                 Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)    4.8957542  0.6178206   7.924  2.3e-15 ***
## Elevation     -0.0018282  0.0006837  -2.674 0.007496 ** 
## Age           -0.3918512  0.1135936  -3.450 0.000561 ***
## Precipitation -0.0010039  0.0002904  -3.457 0.000546 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 414.81  on 42  degrees of freedom
## Residual deviance: 341.86  on 39  degrees of freedom
## AIC: 419.74
## 
## Number of Fisher Scoring iterations: 8</code></pre>
</div>
<div id="model-diagnostic-1" class="section level4 hasAnchor">
<h4>Model Diagnostic<a href="module-3-modeling-part-1.html#model-diagnostic-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Again, after fitting the model, we must check for the following assumptions:</p>
<ol style="list-style-type: decimal">
<li><p>Linearity - use component-residual plots to assess</p></li>
<li><p>Independence - we will assume the data observations are independent</p></li>
<li><p>Proper fitting distribution- we can check this with a scatterplot of model DEVIANCE residuals vs. linear predictor</p></li>
<li><p>No multicollinearity - we can check this with variance inflation factors</p></li>
</ol>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="module-3-modeling-part-1.html#cb194-1" tabindex="-1"></a><span class="co"># check assumptions</span></span>
<span id="cb194-2"><a href="module-3-modeling-part-1.html#cb194-2" tabindex="-1"></a><span class="co"># 1. linearity</span></span>
<span id="cb194-3"><a href="module-3-modeling-part-1.html#cb194-3" tabindex="-1"></a><span class="fu">crPlot</span>(glm.poisson, <span class="at">variable=</span><span class="st">&quot;Elevation&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="module-3-modeling-part-1.html#cb195-1" tabindex="-1"></a><span class="fu">crPlot</span>(glm.poisson, <span class="at">variable=</span><span class="st">&quot;Age&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-36-2.png" width="672" /></p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="module-3-modeling-part-1.html#cb196-1" tabindex="-1"></a><span class="fu">crPlot</span>(glm.poisson, <span class="at">variable=</span><span class="st">&quot;Precipitation&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-36-3.png" width="672" /></p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="module-3-modeling-part-1.html#cb197-1" tabindex="-1"></a><span class="co"># 3. proper fitting distribution</span></span>
<span id="cb197-2"><a href="module-3-modeling-part-1.html#cb197-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">resid</span>(glm.poisson)<span class="sc">~</span><span class="fu">predict</span>(glm.poisson)) <span class="co"># fan pattern - likely due to the overdispersion</span></span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-36-4.png" width="672" /></p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="module-3-modeling-part-1.html#cb198-1" tabindex="-1"></a><span class="co"># 4. no multicollinearity</span></span>
<span id="cb198-2"><a href="module-3-modeling-part-1.html#cb198-2" tabindex="-1"></a><span class="fu">vif</span>(glm.poisson)</span></code></pre></div>
<pre><code>##     Elevation           Age Precipitation 
##      1.193796      1.308440      1.397856</code></pre>
</div>
<div id="addressing-overdispersion-in-poisson" class="section level4 hasAnchor">
<h4>Addressing Overdispersion in Poisson<a href="module-3-modeling-part-1.html#addressing-overdispersion-in-poisson" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The Poisson model assumes that the variance is equal to the mean (i.e., E(Y) = VAR(Y)). If the variance is greater than what is assumed by the model, it is called ‘overdispersion’. We can estimate overdispersion as deviance/df residuals. If this ratio is greater than 1 we have overdispersion, if it is (roughly) 1 we have equidispersion, and if it is less than 1, we have underdispersion.</p>
<p>If we have overdispersion of the Poisson, we can try to fit a quasipoisson model instead which will estimate the dispersion parameter instead of assuming it to be 1.</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="module-3-modeling-part-1.html#cb200-1" tabindex="-1"></a><span class="co"># estimate dispersion</span></span>
<span id="cb200-2"><a href="module-3-modeling-part-1.html#cb200-2" tabindex="-1"></a>glm.poisson<span class="sc">$</span>deviance<span class="sc">/</span>glm.poisson<span class="sc">$</span>df.residual</span></code></pre></div>
<pre><code>## [1] 8.765551</code></pre>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="module-3-modeling-part-1.html#cb202-1" tabindex="-1"></a><span class="co"># fit quasi-poisson to address overdispersion</span></span>
<span id="cb202-2"><a href="module-3-modeling-part-1.html#cb202-2" tabindex="-1"></a>glm.qpoisson <span class="ot">=</span> <span class="fu">glm</span>(Faramea.occidentalis <span class="sc">~</span> Elevation<span class="sc">+</span>Age<span class="sc">+</span>Precipitation,</span>
<span id="cb202-3"><a href="module-3-modeling-part-1.html#cb202-3" tabindex="-1"></a>                  <span class="at">data =</span> faramea,</span>
<span id="cb202-4"><a href="module-3-modeling-part-1.html#cb202-4" tabindex="-1"></a>                  <span class="at">family =</span> quasipoisson)</span>
<span id="cb202-5"><a href="module-3-modeling-part-1.html#cb202-5" tabindex="-1"></a><span class="fu">summary</span>(glm.qpoisson)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = Faramea.occidentalis ~ Elevation + Age + Precipitation, 
##     family = quasipoisson, data = faramea)
## 
## Coefficients:
##                Estimate Std. Error t value Pr(&gt;|t|)  
## (Intercept)    4.895754   2.159927   2.267    0.029 *
## Elevation     -0.001828   0.002390  -0.765    0.449  
## Age           -0.391851   0.397128  -0.987    0.330  
## Precipitation -0.001004   0.001015  -0.989    0.329  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasipoisson family taken to be 12.22231)
## 
##     Null deviance: 414.81  on 42  degrees of freedom
## Residual deviance: 341.86  on 39  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 8</code></pre>
</div>
<div id="modelin-rates-with-poisson" class="section level4 hasAnchor">
<h4>Modelin Rates with Poisson<a href="module-3-modeling-part-1.html#modelin-rates-with-poisson" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>When modeling count data, there are scenarios where the counts observed may vary due to differences in the length of observation periods, the area covered, or the number of trials conducted. In such cases, it’s important to account for the differing observation levels using an offset in Poisson models.</p>
<p>**While the Poisson model can seem similar to the binomial distribution, particularly when dealing with rates of “successes”, there is a key distinction to keep in mind. The Poisson model is generally preferred when the count of successes is unbounded—that is, there is no theoretical limit to the number of successes we might observe based on the denominator. However, if the probability of success is extremely low, so much so that the numerator (the count of successes) would never be expected to approach the denominator, the Poisson model remains appropriate. In contrast, the binomial distribution is typically used when there is a fixed upper limit on the number of successes (and the denominator is this upper limit).</p>
<p>We will briefly review an example here where we have a rate instead of a count outcome. In this example, we will look at the dicentric dataset from faraway. The outcome of interest is the number of chromosomal abnormalities and the predictor of interest is the dose amount and dose rate. However, we also want to account for the number of cells observed as we would expect to see more abnormalities if we observed more cells. We will treat number of cells as our offset variable.</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="module-3-modeling-part-1.html#cb204-1" tabindex="-1"></a><span class="co"># load in the data</span></span>
<span id="cb204-2"><a href="module-3-modeling-part-1.html#cb204-2" tabindex="-1"></a><span class="fu">data</span>(dicentric)</span>
<span id="cb204-3"><a href="module-3-modeling-part-1.html#cb204-3" tabindex="-1"></a></span>
<span id="cb204-4"><a href="module-3-modeling-part-1.html#cb204-4" tabindex="-1"></a><span class="co"># view description of the data</span></span>
<span id="cb204-5"><a href="module-3-modeling-part-1.html#cb204-5" tabindex="-1"></a>?dicentric</span>
<span id="cb204-6"><a href="module-3-modeling-part-1.html#cb204-6" tabindex="-1"></a></span>
<span id="cb204-7"><a href="module-3-modeling-part-1.html#cb204-7" tabindex="-1"></a><span class="co"># now let&#39;s look at the model without accounting for # of cells</span></span>
<span id="cb204-8"><a href="module-3-modeling-part-1.html#cb204-8" tabindex="-1"></a>dicentric<span class="sc">$</span>dose_fac <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(dicentric<span class="sc">$</span>doseamt)</span>
<span id="cb204-9"><a href="module-3-modeling-part-1.html#cb204-9" tabindex="-1"></a>mod_pois <span class="ot">&lt;-</span> <span class="fu">glm</span>(ca <span class="sc">~</span> <span class="fu">log</span>(doserate)<span class="sc">+</span>dose_fac, <span class="at">data=</span>dicentric, <span class="at">family=</span>poisson)</span>
<span id="cb204-10"><a href="module-3-modeling-part-1.html#cb204-10" tabindex="-1"></a><span class="fu">summary</span>(mod_pois) <span class="co"># clearly overdispersed - we are not properly accounting for the variability</span></span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = ca ~ log(doserate) + dose_fac, family = poisson, 
##     data = dicentric)
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)    4.53018    0.03439 131.740   &lt;2e-16 ***
## log(doserate)  0.20443    0.01662  12.301   &lt;2e-16 ***
## dose_fac2.5   -0.02830    0.04857  -0.583     0.56    
## dose_fac5      0.59294    0.04249  13.955   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 916.13  on 26  degrees of freedom
## Residual deviance: 460.69  on 23  degrees of freedom
## AIC: 644.1
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="module-3-modeling-part-1.html#cb206-1" tabindex="-1"></a><span class="co"># include log(cells) as covariate</span></span>
<span id="cb206-2"><a href="module-3-modeling-part-1.html#cb206-2" tabindex="-1"></a>mod_pois <span class="ot">&lt;-</span> <span class="fu">glm</span>(ca <span class="sc">~</span> <span class="fu">log</span>(cells)<span class="sc">+</span><span class="fu">log</span>(doserate)<span class="sc">+</span>dose_fac, <span class="at">data=</span>dicentric, <span class="at">family=</span>poisson) </span>
<span id="cb206-3"><a href="module-3-modeling-part-1.html#cb206-3" tabindex="-1"></a><span class="fu">summary</span>(mod_pois) <span class="co"># close to 1, can fix at 1 and treat as offset</span></span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = ca ~ log(cells) + log(doserate) + dose_fac, family = poisson, 
##     data = dicentric)
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)   -3.08452    0.38201  -8.075 6.77e-16 ***
## log(cells)     1.04271    0.05158  20.217  &lt; 2e-16 ***
## log(doserate)  0.21509    0.01674  12.850  &lt; 2e-16 ***
## dose_fac2.5    1.72742    0.10226  16.892  &lt; 2e-16 ***
## dose_fac5      2.89486    0.12107  23.910  &lt; 2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 916.127  on 26  degrees of freedom
## Residual deviance:  42.089  on 22  degrees of freedom
## AIC: 227.5
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="module-3-modeling-part-1.html#cb208-1" tabindex="-1"></a><span class="co"># include log(cells) as offset</span></span>
<span id="cb208-2"><a href="module-3-modeling-part-1.html#cb208-2" tabindex="-1"></a>mod_pois <span class="ot">&lt;-</span> <span class="fu">glm</span>(ca <span class="sc">~</span> <span class="fu">offset</span>(<span class="fu">log</span>(cells))<span class="sc">+</span><span class="fu">log</span>(doserate)<span class="sc">+</span>dose_fac, <span class="at">data=</span>dicentric, <span class="at">family=</span>poisson)</span>
<span id="cb208-3"><a href="module-3-modeling-part-1.html#cb208-3" tabindex="-1"></a><span class="fu">summary</span>(mod_pois)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = ca ~ offset(log(cells)) + log(doserate) + dose_fac, 
##     family = poisson, data = dicentric)
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)   -2.76958    0.03430  -80.74   &lt;2e-16 ***
## log(doserate)  0.21447    0.01672   12.83   &lt;2e-16 ***
## dose_fac2.5    1.65299    0.04857   34.03   &lt;2e-16 ***
## dose_fac5      2.80095    0.04251   65.89   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 4753.004  on 26  degrees of freedom
## Residual deviance:   42.776  on 23  degrees of freedom
## AIC: 226.18
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>We see that including the offset is important as it can explain a lot of the overdispersion in the model outcome. It is important to note that the GLMs reviewed here are not exhaustive and there are many scenarios you will encounter when these will not work for your data. However, this module is to provide you with the tools necessary to think critically about your data and model assumptions.</p>
<p>If your outcome data is highly overdispersed, it might be better to use more complex models to account for this such as the negative binomial (see below) or the zero-inflated Poisson if you have a large number of zero’s in your outcome.</p>
</div>
</div>
<div id="negative-binomial" class="section level3 hasAnchor">
<h3>Negative Binomial<a href="module-3-modeling-part-1.html#negative-binomial" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will very briefly review negative binomial as it is important for overdispersed count data that often occurs in gene expression data and taxonomic compositions.</p>
<div id="load-the-data" class="section level4 hasAnchor">
<h4>Load the Data<a href="module-3-modeling-part-1.html#load-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The data we are looking at is from a simulated experiment based on RNA-Seq data for differential expression. The first 6 columns are the gene expression of different genes and we have covariates for age, treatment, gender, sex. We additionally want to account for total number of reads as an offset.</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="module-3-modeling-part-1.html#cb210-1" tabindex="-1"></a><span class="co"># load in dataset rnaseq</span></span>
<span id="cb210-2"><a href="module-3-modeling-part-1.html#cb210-2" tabindex="-1"></a><span class="fu">data</span>(rnaseq)</span>
<span id="cb210-3"><a href="module-3-modeling-part-1.html#cb210-3" tabindex="-1"></a></span>
<span id="cb210-4"><a href="module-3-modeling-part-1.html#cb210-4" tabindex="-1"></a><span class="co"># description of dataset</span></span>
<span id="cb210-5"><a href="module-3-modeling-part-1.html#cb210-5" tabindex="-1"></a>?rnaseq</span>
<span id="cb210-6"><a href="module-3-modeling-part-1.html#cb210-6" tabindex="-1"></a></span>
<span id="cb210-7"><a href="module-3-modeling-part-1.html#cb210-7" tabindex="-1"></a><span class="co"># view first 6 rows of dataset</span></span>
<span id="cb210-8"><a href="module-3-modeling-part-1.html#cb210-8" tabindex="-1"></a><span class="fu">head</span>(rnaseq)</span></code></pre></div>
<pre><code>##    X1  X2 X3  X4  X5 X6 totalReads treatment gender age
## 1 295  65 19 114  54 20   28317494         0      0  57
## 2 213 126 12  96  50  4   20015549         0      0  67
## 3 322 147 23 245  42 19   35318251         0      1  37
## 4 184  35  9 134  74  8   20421437         0      0  59
## 5 376 104  4  74  39  7   21940693         0      1  57
## 6 490  97 18 282 159 24   48645477         0      1  50</code></pre>
</div>
<div id="fitting-the-negative-binomial-regression" class="section level4 hasAnchor">
<h4>Fitting the Negative Binomial Regression<a href="module-3-modeling-part-1.html#fitting-the-negative-binomial-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First, we will fit a Poisson regression to show how overdispersed gene expression data can be and how a negative binomial is more accommodating of this kind of data. We can fit this model with glm.nb() which will estimate the extra parameter, theta for us.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="module-3-modeling-part-1.html#cb212-1" tabindex="-1"></a><span class="co"># treat X1 as outcome and fit poisson</span></span>
<span id="cb212-2"><a href="module-3-modeling-part-1.html#cb212-2" tabindex="-1"></a>mod_pois <span class="ot">&lt;-</span> <span class="fu">glm</span>(X1<span class="sc">~</span><span class="fu">offset</span>(<span class="fu">log</span>(totalReads))<span class="sc">+</span>treatment<span class="sc">+</span>gender<span class="sc">+</span>age, <span class="at">data=</span>rnaseq, <span class="at">family=</span>poisson)</span>
<span id="cb212-3"><a href="module-3-modeling-part-1.html#cb212-3" tabindex="-1"></a><span class="fu">summary</span>(mod_pois)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = X1 ~ offset(log(totalReads)) + treatment + gender + 
##     age, family = poisson, data = rnaseq)
## 
## Coefficients:
##               Estimate Std. Error  z value Pr(&gt;|z|)    
## (Intercept) -1.171e+01  2.557e-02 -458.043  &lt; 2e-16 ***
## treatment   -7.581e-01  1.020e-02  -74.337  &lt; 2e-16 ***
## gender      -8.130e-02  9.645e-03   -8.429  &lt; 2e-16 ***
## age          2.407e-03  4.912e-04    4.901 9.55e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 10798.2  on 199  degrees of freedom
## Residual deviance:  4890.3  on 196  degrees of freedom
## AIC: 6313.2
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="module-3-modeling-part-1.html#cb214-1" tabindex="-1"></a><span class="co"># dispersion?</span></span>
<span id="cb214-2"><a href="module-3-modeling-part-1.html#cb214-2" tabindex="-1"></a>sum_pois <span class="ot">&lt;-</span> <span class="fu">summary</span>(mod_pois)</span>
<span id="cb214-3"><a href="module-3-modeling-part-1.html#cb214-3" tabindex="-1"></a>sum_pois<span class="sc">$</span>deviance<span class="sc">/</span>sum_pois<span class="sc">$</span>df.residual</span></code></pre></div>
<pre><code>## [1] 24.95035</code></pre>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="module-3-modeling-part-1.html#cb216-1" tabindex="-1"></a><span class="co"># fit negative binomial</span></span>
<span id="cb216-2"><a href="module-3-modeling-part-1.html#cb216-2" tabindex="-1"></a>mod_nb <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(X1<span class="sc">~</span><span class="fu">offset</span>(<span class="fu">log</span>(totalReads))<span class="sc">+</span>treatment<span class="sc">+</span>gender<span class="sc">+</span>age, <span class="at">data=</span>rnaseq)</span>
<span id="cb216-3"><a href="module-3-modeling-part-1.html#cb216-3" tabindex="-1"></a><span class="fu">summary</span>(mod_nb)</span></code></pre></div>
<pre><code>## 
## Call:
## glm.nb(formula = X1 ~ offset(log(totalReads)) + treatment + gender + 
##     age, data = rnaseq, init.theta = 9.319170881, link = log)
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept) -11.777840   0.126981 -92.753   &lt;2e-16 ***
## treatment    -0.767008   0.047530 -16.137   &lt;2e-16 ***
## gender       -0.043007   0.047568  -0.904    0.366    
## age           0.003463   0.002420   1.431    0.152    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for Negative Binomial(9.3192) family taken to be 1)
## 
##     Null deviance: 460.37  on 199  degrees of freedom
## Residual deviance: 203.51  on 196  degrees of freedom
## AIC: 2243.7
## 
## Number of Fisher Scoring iterations: 1
## 
## 
##               Theta:  9.319 
##           Std. Err.:  0.965 
## 
##  2 x log-likelihood:  -2233.659</code></pre>
</div>
<div id="bonus-beta-regression" class="section level4 hasAnchor">
<h4>Bonus: Beta Regression<a href="module-3-modeling-part-1.html#bonus-beta-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Beta regression is useful for modeling outcomes bounded between 0 and 1, such as proportions. However, it should be used when the denominator of the proportion is unknown. If the denominator is known, binomial regression is more appropriate.</p>
<p>Additionally, beta regression cannot handle outcomes exactly equal to 0 or 1; all values must fall strictly between these bounds.</p>
<p>For this example, we will look at whether there exists an association between DNA methylation at a specific CpG site and gender, age, and smoking status.</p>
</div>
<div id="load-the-data-1" class="section level4 hasAnchor">
<h4>Load the Data<a href="module-3-modeling-part-1.html#load-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The dataset we will use is titled ‘smoker_Epigenetic_df.csv’ in our folder.</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="module-3-modeling-part-1.html#cb218-1" tabindex="-1"></a><span class="co"># first, let&#39;s load in the data</span></span>
<span id="cb218-2"><a href="module-3-modeling-part-1.html#cb218-2" tabindex="-1"></a>smok_dat <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;module3_Smoker_Epigenetic_df.csv&#39;</span>) <span class="co"># smoker_epigenetic_Df</span></span>
<span id="cb218-3"><a href="module-3-modeling-part-1.html#cb218-3" tabindex="-1"></a></span>
<span id="cb218-4"><a href="module-3-modeling-part-1.html#cb218-4" tabindex="-1"></a><span class="co"># view data</span></span>
<span id="cb218-5"><a href="module-3-modeling-part-1.html#cb218-5" tabindex="-1"></a><span class="fu">head</span>(smok_dat)</span></code></pre></div>
<pre><code>##          GSM Smoking.Status Gender Age cg00050873 cg00212031 cg00213748
## 1 GSM1051525        current      f  67  0.6075634  0.4228427  0.3724549
## 2 GSM1051526        current      f  49  0.3450542  0.5686615  0.5005995
## 3 GSM1051527        current      f  53  0.3213497  0.3609091  0.3527315
## 4 GSM1051528        current      f  62  0.2772675  0.3044371  0.4752352
## 5 GSM1051529          never      f  33  0.4135991  0.1312511  0.3675446
## 6 GSM1051530        current      f  59  0.6228599  0.5016849  0.2632270
##   cg00214611 cg00455876 cg01707559 cg02004872 cg02011394 cg02050847 cg02233190
## 1  0.6215619  0.2907773  0.2671431  0.1791439  0.4802517  0.3276078  0.2411204
## 2  0.4986067  0.3745909  0.1902743  0.1559775  0.4180809  0.3464627  0.1754907
## 3  0.3738240  0.2306740  0.3147052  0.1057448  0.6151030  0.2375392  0.2464092
## 4  0.4862581  0.2951815  0.2957931  0.1112862  0.3010196  0.3045353  0.1770279
## 5  0.7611667  0.2357703  0.2505265  0.1691084  0.3929746  0.3062257  0.3017014
## 6  0.4157459  0.4751891  0.2539041  0.2607587  0.5097921  0.4052457  0.3852716
##   cg02494853  cg02839557 cg02842889 cg03052502 cg03155755 cg03244189 cg03443143
## 1 0.06706958 0.246993368  0.4692396  0.4002466  0.4150313  0.2214331  0.4758258
## 2 0.04693889 0.236742313  0.3074666  0.3770313  0.3973715  0.2171221  0.5444690
## 3 0.03823712 0.244611725  0.3577526  0.3050442  0.5212775  0.1850495  0.5370600
## 4 0.02671625 0.001641439  0.4457390  0.2714746  0.4344920  0.1654187  0.5079167
## 5 0.03701636 0.334319727  0.3950396  0.3265530  0.4300966  0.1811352  0.4054791
## 6 0.02583463 0.309210202  0.3218573  0.5333670  0.5715522  0.2109749  0.3778239
##   cg03683899 cg03695421 cg03706273
## 1  0.2077242  0.2091974 0.12998255
## 2  0.1844462  0.1937732 0.09853265
## 3  0.3931231  0.2680030 0.04024808
## 4  0.2812089  0.2178572 0.10151626
## 5  0.3107944  0.2800708 0.07785712
## 6  0.4693609  0.3433317 0.04577912</code></pre>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="module-3-modeling-part-1.html#cb220-1" tabindex="-1"></a><span class="co"># look at structure</span></span>
<span id="cb220-2"><a href="module-3-modeling-part-1.html#cb220-2" tabindex="-1"></a><span class="fu">str</span>(smok_dat)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    683 obs. of  24 variables:
##  $ GSM           : chr  &quot;GSM1051525&quot; &quot;GSM1051526&quot; &quot;GSM1051527&quot; &quot;GSM1051528&quot; ...
##  $ Smoking.Status: chr  &quot;current&quot; &quot;current&quot; &quot;current&quot; &quot;current&quot; ...
##  $ Gender        : chr  &quot; f&quot; &quot; f&quot; &quot; f&quot; &quot; f&quot; ...
##  $ Age           : int  67 49 53 62 33 59 66 51 55 37 ...
##  $ cg00050873    : num  0.608 0.345 0.321 0.277 0.414 ...
##  $ cg00212031    : num  0.423 0.569 0.361 0.304 0.131 ...
##  $ cg00213748    : num  0.372 0.501 0.353 0.475 0.368 ...
##  $ cg00214611    : num  0.622 0.499 0.374 0.486 0.761 ...
##  $ cg00455876    : num  0.291 0.375 0.231 0.295 0.236 ...
##  $ cg01707559    : num  0.267 0.19 0.315 0.296 0.251 ...
##  $ cg02004872    : num  0.179 0.156 0.106 0.111 0.169 ...
##  $ cg02011394    : num  0.48 0.418 0.615 0.301 0.393 ...
##  $ cg02050847    : num  0.328 0.346 0.238 0.305 0.306 ...
##  $ cg02233190    : num  0.241 0.175 0.246 0.177 0.302 ...
##  $ cg02494853    : num  0.0671 0.0469 0.0382 0.0267 0.037 ...
##  $ cg02839557    : num  0.24699 0.23674 0.24461 0.00164 0.33432 ...
##  $ cg02842889    : num  0.469 0.307 0.358 0.446 0.395 ...
##  $ cg03052502    : num  0.4 0.377 0.305 0.271 0.327 ...
##  $ cg03155755    : num  0.415 0.397 0.521 0.434 0.43 ...
##  $ cg03244189    : num  0.221 0.217 0.185 0.165 0.181 ...
##  $ cg03443143    : num  0.476 0.544 0.537 0.508 0.405 ...
##  $ cg03683899    : num  0.208 0.184 0.393 0.281 0.311 ...
##  $ cg03695421    : num  0.209 0.194 0.268 0.218 0.28 ...
##  $ cg03706273    : num  0.13 0.0985 0.0402 0.1015 0.0779 ...</code></pre>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="module-3-modeling-part-1.html#cb222-1" tabindex="-1"></a><span class="co"># look at summary</span></span>
<span id="cb222-2"><a href="module-3-modeling-part-1.html#cb222-2" tabindex="-1"></a><span class="fu">summary</span>(smok_dat) <span class="co"># some missingness in our outcome, we will not impute</span></span></code></pre></div>
<pre><code>##      GSM            Smoking.Status        Gender               Age       
##  Length:683         Length:683         Length:683         Min.   :18.00  
##  Class :character   Class :character   Class :character   1st Qu.:47.00  
##  Mode  :character   Mode  :character   Mode  :character   Median :56.00  
##                                                           Mean   :53.82  
##                                                           3rd Qu.:62.00  
##                                                           Max.   :80.00  
##                                                                          
##    cg00050873       cg00212031        cg00213748       cg00214611     
##  Min.   :0.1186   Min.   :0.00695   Min.   :0.0000   Min.   :0.01247  
##  1st Qu.:0.4131   1st Qu.:0.06317   1st Qu.:0.3635   1st Qu.:0.06946  
##  Median :0.5052   Median :0.36554   Median :0.4713   Median :0.41575  
##  Mean   :0.5600   Mean   :0.30960   Mean   :0.5191   Mean   :0.34106  
##  3rd Qu.:0.8144   3rd Qu.:0.45981   3rd Qu.:0.7278   3rd Qu.:0.49745  
##  Max.   :0.8989   Max.   :0.70999   Max.   :0.9236   Max.   :0.80606  
##  NA&#39;s   :62       NA&#39;s   :62        NA&#39;s   :62       NA&#39;s   :62       
##    cg00455876        cg01707559        cg02004872        cg02011394    
##  Min.   :0.05917   Min.   :0.04333   Min.   :0.00262   Min.   :0.0000  
##  1st Qu.:0.29300   1st Qu.:0.11080   1st Qu.:0.04284   1st Qu.:0.4261  
##  Median :0.37968   Median :0.23873   Median :0.14933   Median :0.5157  
##  Mean   :0.44718   Mean   :0.21435   Mean   :0.15542   Mean   :0.6058  
##  3rd Qu.:0.66283   3rd Qu.:0.28061   3rd Qu.:0.24263   3rd Qu.:0.9412  
##  Max.   :0.85443   Max.   :0.46999   Max.   :0.47384   Max.   :0.9792  
##  NA&#39;s   :62        NA&#39;s   :62        NA&#39;s   :62        NA&#39;s   :62      
##    cg02050847        cg02233190        cg02494853        cg02839557     
##  Min.   :0.05234   Min.   :0.00863   Min.   :0.01162   Min.   :0.00000  
##  1st Qu.:0.33963   1st Qu.:0.08838   1st Qu.:0.02865   1st Qu.:0.06384  
##  Median :0.42754   Median :0.25982   Median :0.03695   Median :0.35042  
##  Mean   :0.54369   Mean   :0.23250   Mean   :0.04077   Mean   :0.30088  
##  3rd Qu.:0.95558   3rd Qu.:0.33702   3rd Qu.:0.04677   3rd Qu.:0.45786  
##  Max.   :0.98320   Max.   :0.51173   Max.   :0.28947   Max.   :0.82739  
##  NA&#39;s   :62        NA&#39;s   :62        NA&#39;s   :62        NA&#39;s   :62       
##    cg02842889        cg03052502       cg03155755       cg03244189     
##  Min.   :0.01346   Min.   :0.0000   Min.   :0.2020   Min.   :0.02972  
##  1st Qu.:0.05483   1st Qu.:0.4025   1st Qu.:0.4245   1st Qu.:0.11976  
##  Median :0.39757   Median :0.4940   Median :0.4962   Median :0.20397  
##  Mean   :0.32362   Mean   :0.5907   Mean   :0.5895   Mean   :0.19552  
##  3rd Qu.:0.47385   3rd Qu.:0.9631   3rd Qu.:0.8988   3rd Qu.:0.24921  
##  Max.   :0.85625   Max.   :0.9902   Max.   :0.9696   Max.   :0.54074  
##  NA&#39;s   :62        NA&#39;s   :62       NA&#39;s   :62       NA&#39;s   :62       
##    cg03443143        cg03683899        cg03695421       cg03706273     
##  Min.   :0.06496   Min.   :0.00788   Min.   :0.0949   Min.   :0.01120  
##  1st Qu.:0.40963   1st Qu.:0.06159   1st Qu.:0.2566   1st Qu.:0.03413  
##  Median :0.48314   Median :0.34422   Median :0.3208   Median :0.04961  
##  Mean   :0.56841   Mean   :0.28442   Mean   :0.3978   Mean   :0.05769  
##  3rd Qu.:0.85436   3rd Qu.:0.41866   3rd Qu.:0.5965   3rd Qu.:0.06916  
##  Max.   :0.93589   Max.   :0.65925   Max.   :0.8433   Max.   :0.34380  
##  NA&#39;s   :62        NA&#39;s   :62        NA&#39;s   :62       NA&#39;s   :62</code></pre>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="module-3-modeling-part-1.html#cb224-1" tabindex="-1"></a><span class="co"># look closer at gender and smoking status</span></span>
<span id="cb224-2"><a href="module-3-modeling-part-1.html#cb224-2" tabindex="-1"></a><span class="fu">table</span>(smok_dat<span class="sc">$</span>Gender)</span></code></pre></div>
<pre><code>## 
##   f   F   m   M 
## 440  13 181  49</code></pre>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="module-3-modeling-part-1.html#cb226-1" tabindex="-1"></a><span class="fu">table</span>(smok_dat<span class="sc">$</span>Smoking.Status)</span></code></pre></div>
<pre><code>## 
## current   never 
##     490     193</code></pre>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="module-3-modeling-part-1.html#cb228-1" tabindex="-1"></a><span class="co"># clean gender</span></span>
<span id="cb228-2"><a href="module-3-modeling-part-1.html#cb228-2" tabindex="-1"></a>smok_dat_c <span class="ot">&lt;-</span> smok_dat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Gender=</span><span class="fu">case_when</span>(Gender<span class="sc">==</span><span class="st">&quot; f&quot;</span><span class="sc">~</span><span class="st">&quot;F&quot;</span>,</span>
<span id="cb228-3"><a href="module-3-modeling-part-1.html#cb228-3" tabindex="-1"></a>                                                   Gender<span class="sc">==</span><span class="st">&quot; F&quot;</span><span class="sc">~</span><span class="st">&quot;F&quot;</span>,</span>
<span id="cb228-4"><a href="module-3-modeling-part-1.html#cb228-4" tabindex="-1"></a>                                                   Gender<span class="sc">==</span><span class="st">&quot; m&quot;</span><span class="sc">~</span><span class="st">&quot;M&quot;</span>,</span>
<span id="cb228-5"><a href="module-3-modeling-part-1.html#cb228-5" tabindex="-1"></a>                                                   Gender<span class="sc">==</span><span class="st">&quot; M&quot;</span><span class="sc">~</span><span class="st">&quot;M&quot;</span>)) </span>
<span id="cb228-6"><a href="module-3-modeling-part-1.html#cb228-6" tabindex="-1"></a><span class="fu">table</span>(smok_dat_c<span class="sc">$</span>Gender)</span></code></pre></div>
<pre><code>## 
##   F   M 
## 453 230</code></pre>
</div>
<div id="fitting-beta-regression" class="section level4 hasAnchor">
<h4>Fitting Beta Regression<a href="module-3-modeling-part-1.html#fitting-beta-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To fit beta regression, we need to use the gam() function from the mgcv package in R. We set family=betar() in this function.</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="module-3-modeling-part-1.html#cb230-1" tabindex="-1"></a><span class="co"># look at effect of smoking status, gender, and age on methylation</span></span>
<span id="cb230-2"><a href="module-3-modeling-part-1.html#cb230-2" tabindex="-1"></a>modb <span class="ot">&lt;-</span> <span class="fu">gam</span>(cg00455876<span class="sc">~</span>Smoking.Status<span class="sc">+</span>Gender<span class="sc">+</span>Age, <span class="at">data=</span>smok_dat_c, <span class="at">family=</span><span class="fu">betar</span>())</span>
<span id="cb230-3"><a href="module-3-modeling-part-1.html#cb230-3" tabindex="-1"></a><span class="fu">summary</span>(modb)</span></code></pre></div>
<pre><code>## 
## Family: Beta regression(31.518) 
## Link function: logit 
## 
## Formula:
## cg00455876 ~ Smoking.Status + Gender + Age
## 
## Parametric coefficients:
##                       Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)         -0.7581417  0.0731284 -10.367   &lt;2e-16 ***
## Smoking.Statusnever  0.0655020  0.0328325   1.995    0.046 *  
## GenderM              1.6778920  0.0342145  49.040   &lt;2e-16 ***
## Age                  0.0005844  0.0013288   0.440    0.660    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## 
## R-sq.(adj) =  0.832   Deviance explained = 82.5%
## -REML = -670.56  Scale est. = 1         n = 621</code></pre>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="module-3-modeling-part-1.html#cb232-1" tabindex="-1"></a><span class="co"># check assumptions with gam.check</span></span>
<span id="cb232-2"><a href="module-3-modeling-part-1.html#cb232-2" tabindex="-1"></a><span class="fu">gam.check</span>(modb)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<pre><code>## 
## Method: REML   Optimizer: outer newton
## full convergence after 6 iterations.
## Gradient range [4.752114e-09,4.752114e-09]
## (score -670.5585 &amp; scale 1).
## Hessian positive definite, eigenvalue range [317.2695,317.2695].
## Model rank =  4 / 4</code></pre>
<p>The mgcv package has a useful function, gam.check, that allows us to check all of our usual model assumptions. In the QQ plot, we want the dots to follow roughly along the diagonal line if we assumed the correct distribution (beta) for our data. In the residual vs fitted plot, we are looking for a random scattering of points as usual.</p>
</div>
</div>
<div id="independent-work" class="section level3 hasAnchor">
<h3>Independent Work<a href="module-3-modeling-part-1.html#independent-work" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="exercise-1---baboon-diet-and-social-group-size" class="section level4 hasAnchor">
<h4>Exercise 1 - Baboon Diet and Social Group Size<a href="module-3-modeling-part-1.html#exercise-1---baboon-diet-and-social-group-size" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The baboongut dataset from the microbiomeDataSets contains data on baboons over a 14 year span to look at heritability of the gut microbiome within social groups. Let’s load in this dataset and take a closer look. First, we will filter the data to only keep the first observation for each baboon to make the data cross-sectional as opposed to longitudinal.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="module-3-modeling-part-1.html#cb234-1" tabindex="-1"></a><span class="co"># load in dataset </span></span>
<span id="cb234-2"><a href="module-3-modeling-part-1.html#cb234-2" tabindex="-1"></a>ts <span class="ot">&lt;-</span> <span class="fu">baboongut</span>()</span>
<span id="cb234-3"><a href="module-3-modeling-part-1.html#cb234-3" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">colData</span>(ts) <span class="co"># colData gives us the metadata we need</span></span>
<span id="cb234-4"><a href="module-3-modeling-part-1.html#cb234-4" tabindex="-1"></a></span>
<span id="cb234-5"><a href="module-3-modeling-part-1.html#cb234-5" tabindex="-1"></a><span class="co"># look at description</span></span>
<span id="cb234-6"><a href="module-3-modeling-part-1.html#cb234-6" tabindex="-1"></a>?baboongut</span>
<span id="cb234-7"><a href="module-3-modeling-part-1.html#cb234-7" tabindex="-1"></a></span>
<span id="cb234-8"><a href="module-3-modeling-part-1.html#cb234-8" tabindex="-1"></a><span class="co"># we will only look at first sample per ID (so it is not longitudinal)</span></span>
<span id="cb234-9"><a href="module-3-modeling-part-1.html#cb234-9" tabindex="-1"></a>baboon_first <span class="ot">&lt;-</span> sample_data[<span class="fu">isUnique</span>(sample_data<span class="sc">$</span>baboon_id), ]</span>
<span id="cb234-10"><a href="module-3-modeling-part-1.html#cb234-10" tabindex="-1"></a><span class="fu">dim</span>(baboon_first)</span></code></pre></div>
<pre><code>## [1] 62 35</code></pre>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="module-3-modeling-part-1.html#cb236-1" tabindex="-1"></a><span class="co"># we will only focus on the diet variables</span></span>
<span id="cb236-2"><a href="module-3-modeling-part-1.html#cb236-2" tabindex="-1"></a>baboon_first_c <span class="ot">&lt;-</span> baboon_first <span class="sc">%&gt;%</span></span>
<span id="cb236-3"><a href="module-3-modeling-part-1.html#cb236-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span></span>
<span id="cb236-4"><a href="module-3-modeling-part-1.html#cb236-4" tabindex="-1"></a>                  dplyr<span class="sc">::</span><span class="fu">select</span>(group_size, <span class="fu">contains</span>(<span class="st">&quot;diet_PC&quot;</span>))</span>
<span id="cb236-5"><a href="module-3-modeling-part-1.html#cb236-5" tabindex="-1"></a></span>
<span id="cb236-6"><a href="module-3-modeling-part-1.html#cb236-6" tabindex="-1"></a><span class="fu">dim</span>(baboon_first_c)</span></code></pre></div>
<pre><code>## [1] 62 14</code></pre>
<p>Now, do the following:</p>
<ol style="list-style-type: decimal">
<li>Create a new dataset called ‘baboon_first_c’ that only has ‘group_size’ and the first 7 “diet_PC’ columns. These are the variables you would get had you done a principal component analysis on diet variables to reduce the dimensionality of your data. We will be using these principal components as our new predictors.</li>
</ol>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="module-3-modeling-part-1.html#cb238-1" tabindex="-1"></a><span class="co"># we can use select from tidyverse to do this</span></span>
<span id="cb238-2"><a href="module-3-modeling-part-1.html#cb238-2" tabindex="-1"></a>baboon_first_c <span class="ot">&lt;-</span> baboon_first <span class="sc">%&gt;%</span></span>
<span id="cb238-3"><a href="module-3-modeling-part-1.html#cb238-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span></span>
<span id="cb238-4"><a href="module-3-modeling-part-1.html#cb238-4" tabindex="-1"></a>                  dplyr<span class="sc">::</span><span class="fu">select</span>(group_size, diet_PC1<span class="sc">:</span>diet_PC7)</span>
<span id="cb238-5"><a href="module-3-modeling-part-1.html#cb238-5" tabindex="-1"></a></span>
<span id="cb238-6"><a href="module-3-modeling-part-1.html#cb238-6" tabindex="-1"></a><span class="fu">dim</span>(baboon_first_c)</span></code></pre></div>
<pre><code>## [1] 62  8</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Fit a linear regression model with the lm() function with group_size as the outcome and the principal components of diet variables (diet_PC) as your predictors.</li>
</ol>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="module-3-modeling-part-1.html#cb240-1" tabindex="-1"></a><span class="co"># fit model </span></span>
<span id="cb240-2"><a href="module-3-modeling-part-1.html#cb240-2" tabindex="-1"></a>mod1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(group_size<span class="sc">~</span>., baboon_first_c)</span>
<span id="cb240-3"><a href="module-3-modeling-part-1.html#cb240-3" tabindex="-1"></a><span class="fu">summary</span>(mod1)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = group_size ~ ., data = baboon_first_c)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -48.092 -12.943  -2.438  15.847  44.974 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) 71.22209    3.16188  22.525  &lt; 2e-16 ***
## diet_PC1     0.07252    0.09040   0.802   0.4259    
## diet_PC2     0.17070    0.23158   0.737   0.4643    
## diet_PC3    -0.52114    0.32445  -1.606   0.1141    
## diet_PC4     0.12835    0.40988   0.313   0.7554    
## diet_PC5     0.80878    0.73445   1.101   0.2757    
## diet_PC6     1.13856    0.53488   2.129   0.0379 *  
## diet_PC7     4.36778    0.98211   4.447 4.37e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 21.67 on 54 degrees of freedom
## Multiple R-squared:  0.4013, Adjusted R-squared:  0.3237 
## F-statistic:  5.17 on 7 and 54 DF,  p-value: 0.0001502</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Check the model assumptions of this linear regression model. What violations can you find? Hint: think about all assumptions, not just those that we can test/visualize.</li>
</ol>
<p>Can you think of how you would try to fix these violations?</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="module-3-modeling-part-1.html#cb242-1" tabindex="-1"></a><span class="co"># multicollinearity?</span></span>
<span id="cb242-2"><a href="module-3-modeling-part-1.html#cb242-2" tabindex="-1"></a><span class="fu">vif</span>(mod1)</span></code></pre></div>
<pre><code>## diet_PC1 diet_PC2 diet_PC3 diet_PC4 diet_PC5 diet_PC6 diet_PC7 
## 1.051217 1.084132 1.092934 1.177352 1.814580 1.386233 1.504968</code></pre>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="module-3-modeling-part-1.html#cb244-1" tabindex="-1"></a><span class="co"># linearity?</span></span>
<span id="cb244-2"><a href="module-3-modeling-part-1.html#cb244-2" tabindex="-1"></a><span class="fu">pairs</span>(baboon_first_c)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-46-1.png" width="672" /></p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="module-3-modeling-part-1.html#cb245-1" tabindex="-1"></a><span class="co"># normality?</span></span>
<span id="cb245-2"><a href="module-3-modeling-part-1.html#cb245-2" tabindex="-1"></a><span class="fu">qqPlot</span>(mod1)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-46-2.png" width="672" /></p>
<pre><code>## sample_12052-TAGATCCTCGGA-408 sample_12050-TATGCCAGAGAT-407 
##                            38                            52</code></pre>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="module-3-modeling-part-1.html#cb247-1" tabindex="-1"></a><span class="co"># constant variance?</span></span>
<span id="cb247-2"><a href="module-3-modeling-part-1.html#cb247-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">resid</span>(mod1)<span class="sc">~</span><span class="fu">predict</span>(mod1))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-46-3.png" width="672" /></p>
<p>The key violation here would likely be that we do not have independent observations as those within the same social group are more likely to be similar to one another in how they eat (and in this case, will have the same outcome). Other potential violations would be that there seems to be a bit of increasing variance in model residuals which could be because our outcome is a discrete count and Poisson might be more appropriate.</p>
</div>
<div id="exercise-2---healthcare-demand-in-u.s.a" class="section level4 hasAnchor">
<h4>Exercise 2 - Healthcare demand in U.S.A<a href="module-3-modeling-part-1.html#exercise-2---healthcare-demand-in-u.s.a" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For this example we will look at the demand of healthcare in the U.S.A based on certain risk factors such as age, sex, income, etc. We will first load in the dataset, ‘healthcare_demand_nmes1988.csv’ from our folder.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="module-3-modeling-part-1.html#cb248-1" tabindex="-1"></a><span class="co"># 1. Load in the dataset.</span></span>
<span id="cb248-2"><a href="module-3-modeling-part-1.html#cb248-2" tabindex="-1"></a>health <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;module3_healthcare_demand_nmes1988.csv&#39;</span>) <span class="co"># Healthcare demand dataset</span></span>
<span id="cb248-3"><a href="module-3-modeling-part-1.html#cb248-3" tabindex="-1"></a></span>
<span id="cb248-4"><a href="module-3-modeling-part-1.html#cb248-4" tabindex="-1"></a><span class="co"># 2. view data</span></span>
<span id="cb248-5"><a href="module-3-modeling-part-1.html#cb248-5" tabindex="-1"></a><span class="fu">str</span>(health)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    4406 obs. of  18 variables:
##  $ visits   : int  5 1 13 16 3 17 9 3 1 0 ...
##  $ nvisits  : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ ovisits  : int  0 2 0 5 0 0 0 0 0 0 ...
##  $ novisits : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ emergency: int  0 2 3 1 0 0 0 0 0 0 ...
##  $ hospital : int  1 0 3 1 0 0 0 0 0 0 ...
##  $ health   : chr  &quot;average&quot; &quot;average&quot; &quot;poor&quot; &quot;poor&quot; ...
##  $ chronic  : int  2 2 4 2 2 5 0 0 0 0 ...
##  $ adl      : chr  &quot;normal&quot; &quot;normal&quot; &quot;limited&quot; &quot;limited&quot; ...
##  $ region   : chr  &quot;other&quot; &quot;other&quot; &quot;other&quot; &quot;other&quot; ...
##  $ age      : num  6.9 7.4 6.6 7.6 7.9 6.6 7.5 8.7 7.3 7.8 ...
##  $ gender   : chr  &quot;male&quot; &quot;female&quot; &quot;female&quot; &quot;male&quot; ...
##  $ married  : chr  &quot;yes&quot; &quot;yes&quot; &quot;no&quot; &quot;yes&quot; ...
##  $ school   : int  6 10 10 3 6 7 8 8 8 8 ...
##  $ income   : num  2.881 2.748 0.653 0.659 0.659 ...
##  $ employed : chr  &quot;yes&quot; &quot;no&quot; &quot;no&quot; &quot;no&quot; ...
##  $ insurance: chr  &quot;yes&quot; &quot;yes&quot; &quot;no&quot; &quot;yes&quot; ...
##  $ medicaid : chr  &quot;no&quot; &quot;no&quot; &quot;yes&quot; &quot;no&quot; ...</code></pre>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="module-3-modeling-part-1.html#cb250-1" tabindex="-1"></a><span class="fu">head</span>(health)</span></code></pre></div>
<pre><code>##   visits nvisits ovisits novisits emergency hospital  health chronic     adl
## 1      5       0       0        0         0        1 average       2  normal
## 2      1       0       2        0         2        0 average       2  normal
## 3     13       0       0        0         3        3    poor       4 limited
## 4     16       0       5        0         1        1    poor       2 limited
## 5      3       0       0        0         0        0 average       2 limited
## 6     17       0       0        0         0        0    poor       5 limited
##   region age gender married school income employed insurance medicaid
## 1  other 6.9   male     yes      6 2.8810      yes       yes       no
## 2  other 7.4 female     yes     10 2.7478       no       yes       no
## 3  other 6.6 female      no     10 0.6532       no        no      yes
## 4  other 7.6   male     yes      3 0.6588       no       yes       no
## 5  other 7.9 female     yes      6 0.6588       no       yes       no
## 6  other 6.6 female      no      7 0.3301       no        no      yes</code></pre>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="module-3-modeling-part-1.html#cb252-1" tabindex="-1"></a><span class="co"># fix income (notice that some incomes are negative, we will set those to 0)</span></span>
<span id="cb252-2"><a href="module-3-modeling-part-1.html#cb252-2" tabindex="-1"></a>health<span class="sc">$</span>income[health<span class="sc">$</span>income<span class="sc">&lt;</span><span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Choose an appropriate regression model if visits is our outcome and our predictors are health, chronic, adl, income, gender, age, insurance, medicaid. Additionally include an interaction term between income and gender. Fit this model to the data.</li>
</ol>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="module-3-modeling-part-1.html#cb253-1" tabindex="-1"></a><span class="co"># we will choose a Poisson regression given that the outcome is a discrete count</span></span>
<span id="cb253-2"><a href="module-3-modeling-part-1.html#cb253-2" tabindex="-1"></a>glm_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(visits<span class="sc">~</span>health<span class="sc">+</span>chronic<span class="sc">+</span>adl<span class="sc">+</span>income<span class="sc">*</span>gender<span class="sc">+</span>age<span class="sc">+</span>insurance<span class="sc">+</span>medicaid, <span class="at">family=</span><span class="st">&quot;poisson&quot;</span>, <span class="at">data=</span>health)</span>
<span id="cb253-3"><a href="module-3-modeling-part-1.html#cb253-3" tabindex="-1"></a><span class="fu">summary</span>(glm_mod)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = visits ~ health + chronic + adl + income * gender + 
##     age + insurance + medicaid, family = &quot;poisson&quot;, data = health)
## 
## Coefficients:
##                    Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)        1.699105   0.085126  19.960  &lt; 2e-16 ***
## healthexcellent   -0.353853   0.030311 -11.674  &lt; 2e-16 ***
## healthpoor         0.259091   0.018087  14.325  &lt; 2e-16 ***
## chronic            0.163046   0.004534  35.964  &lt; 2e-16 ***
## adlnormal         -0.089349   0.016605  -5.381 7.41e-08 ***
## income            -0.003374   0.003104  -1.087    0.277    
## gendermale        -0.137261   0.017031  -8.060 7.66e-16 ***
## age               -0.062546   0.010478  -5.969 2.38e-09 ***
## insuranceyes       0.378897   0.018923  20.023  &lt; 2e-16 ***
## medicaidyes        0.248354   0.024639  10.080  &lt; 2e-16 ***
## income:gendermale  0.017089   0.004123   4.145 3.40e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 26943  on 4405  degrees of freedom
## Residual deviance: 23818  on 4395  degrees of freedom
## AIC: 36615
## 
## Number of Fisher Scoring iterations: 5</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Interpret the interaction term between income and gender</li>
</ol>
<p>Income has a larger association with nummber of visits for males such that males with higher income are likely to have more visits while income does not have an effect on visits for females.</p>
<ol start="3" style="list-style-type: decimal">
<li>Check model assumptions (also look for overdispersion if appropriate for the model you chose) - note that you cannot use crPlot since we included an interaction so we can assume the linearity assumption is met for now.</li>
</ol>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="module-3-modeling-part-1.html#cb255-1" tabindex="-1"></a><span class="co"># constant variance of deviance residuals?</span></span>
<span id="cb255-2"><a href="module-3-modeling-part-1.html#cb255-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">resid</span>(glm_mod)<span class="sc">~</span><span class="fu">predict</span>(glm_mod)) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="module-3-modeling-part-1.html#cb256-1" tabindex="-1"></a><span class="co"># overdispersed?</span></span>
<span id="cb256-2"><a href="module-3-modeling-part-1.html#cb256-2" tabindex="-1"></a>glm_summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(glm_mod)</span>
<span id="cb256-3"><a href="module-3-modeling-part-1.html#cb256-3" tabindex="-1"></a>glm_summary<span class="sc">$</span>deviance<span class="sc">/</span>glm_summary<span class="sc">$</span>df.residual <span class="co"># overdispersed</span></span></code></pre></div>
<pre><code>## [1] 5.419233</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Fix any model violations you may have noticed.</li>
</ol>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="module-3-modeling-part-1.html#cb258-1" tabindex="-1"></a><span class="co"># we will log transform income (need to add 0.1 since we cannot take log of 0)</span></span>
<span id="cb258-2"><a href="module-3-modeling-part-1.html#cb258-2" tabindex="-1"></a>health<span class="sc">$</span>income_log  <span class="ot">&lt;-</span> <span class="fu">log</span>(health<span class="sc">$</span>income<span class="fl">+0.1</span>)</span>
<span id="cb258-3"><a href="module-3-modeling-part-1.html#cb258-3" tabindex="-1"></a><span class="co"># Fitting quasipoisson to account for overdisperion</span></span>
<span id="cb258-4"><a href="module-3-modeling-part-1.html#cb258-4" tabindex="-1"></a>glm_mod1 <span class="ot">&lt;-</span> <span class="fu">glm</span>(visits<span class="sc">~</span>health<span class="sc">+</span>chronic<span class="sc">+</span>adl<span class="sc">+</span>income_log<span class="sc">*</span>gender<span class="sc">+</span>age<span class="sc">+</span>insurance<span class="sc">+</span>medicaid, <span class="at">family=</span><span class="st">&quot;quasipoisson&quot;</span>, <span class="at">data=</span>health)</span>
<span id="cb258-5"><a href="module-3-modeling-part-1.html#cb258-5" tabindex="-1"></a><span class="fu">summary</span>(glm_mod1)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = visits ~ health + chronic + adl + income_log * 
##     gender + age + insurance + medicaid, family = &quot;quasipoisson&quot;, 
##     data = health)
## 
## Coefficients:
##                       Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)            1.68935    0.22585   7.480 8.93e-14 ***
## healthexcellent       -0.35470    0.08038  -4.413 1.05e-05 ***
## healthpoor             0.26340    0.04803   5.484 4.40e-08 ***
## chronic                0.16305    0.01203  13.555  &lt; 2e-16 ***
## adlnormal             -0.08755    0.04408  -1.986 0.047067 *  
## income_log            -0.01862    0.02621  -0.710 0.477459    
## gendermale            -0.16838    0.04704  -3.579 0.000348 ***
## age                   -0.06075    0.02784  -2.182 0.029129 *  
## insuranceyes           0.37319    0.05050   7.390 1.74e-13 ***
## medicaidyes            0.24667    0.06602   3.736 0.000189 ***
## income_log:gendermale  0.10583    0.04323   2.448 0.014410 *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasipoisson family taken to be 7.032885)
## 
##     Null deviance: 26943  on 4405  degrees of freedom
## Residual deviance: 23794  on 4395  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 5</code></pre>
</div>
<div id="advanced-exercise---multivariate-count-data" class="section level4 hasAnchor">
<h4>Advanced exercise - Multivariate count data<a href="module-3-modeling-part-1.html#advanced-exercise---multivariate-count-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="module-3-modeling-part-1.html#cb260-1" tabindex="-1"></a><span class="co"># we will log transform income (need to add 0.1 since we cannot take log of 0)</span></span>
<span id="cb260-2"><a href="module-3-modeling-part-1.html#cb260-2" tabindex="-1"></a>health<span class="sc">$</span>income_log  <span class="ot">&lt;-</span> <span class="fu">log</span>(health<span class="sc">$</span>income<span class="fl">+0.1</span>)</span>
<span id="cb260-3"><a href="module-3-modeling-part-1.html#cb260-3" tabindex="-1"></a><span class="co"># Fitting quasipoisson to account for overdisperion</span></span>
<span id="cb260-4"><a href="module-3-modeling-part-1.html#cb260-4" tabindex="-1"></a>glm_mod1 <span class="ot">&lt;-</span> <span class="fu">glm</span>(visits<span class="sc">~</span>health<span class="sc">+</span>chronic<span class="sc">+</span>adl<span class="sc">+</span>income_log<span class="sc">*</span>gender<span class="sc">+</span>age<span class="sc">+</span>insurance<span class="sc">+</span>medicaid, <span class="at">family=</span><span class="st">&quot;quasipoisson&quot;</span>, <span class="at">data=</span>health)</span>
<span id="cb260-5"><a href="module-3-modeling-part-1.html#cb260-5" tabindex="-1"></a><span class="fu">summary</span>(glm_mod1)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = visits ~ health + chronic + adl + income_log * 
##     gender + age + insurance + medicaid, family = &quot;quasipoisson&quot;, 
##     data = health)
## 
## Coefficients:
##                       Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)            1.68935    0.22585   7.480 8.93e-14 ***
## healthexcellent       -0.35470    0.08038  -4.413 1.05e-05 ***
## healthpoor             0.26340    0.04803   5.484 4.40e-08 ***
## chronic                0.16305    0.01203  13.555  &lt; 2e-16 ***
## adlnormal             -0.08755    0.04408  -1.986 0.047067 *  
## income_log            -0.01862    0.02621  -0.710 0.477459    
## gendermale            -0.16838    0.04704  -3.579 0.000348 ***
## age                   -0.06075    0.02784  -2.182 0.029129 *  
## insuranceyes           0.37319    0.05050   7.390 1.74e-13 ***
## medicaidyes            0.24667    0.06602   3.736 0.000189 ***
## income_log:gendermale  0.10583    0.04323   2.448 0.014410 *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasipoisson family taken to be 7.032885)
## 
##     Null deviance: 26943  on 4405  degrees of freedom
## Residual deviance: 23794  on 4395  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 5</code></pre>
</div>
<div id="advanced-exercise---multivariate-count-data-1" class="section level4 hasAnchor">
<h4>Advanced exercise - Multivariate count data<a href="module-3-modeling-part-1.html#advanced-exercise---multivariate-count-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Often in gene expression data or compositional data, we have multiple counts. The package MGLM in R offers functions to fit multivariate count models such as the multinomial model and more advanced models to handle overdispersion of these counts.</p>
<p>Let’s look again at the rna seq dataset we used for the negative binomial example.</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="module-3-modeling-part-1.html#cb262-1" tabindex="-1"></a><span class="co"># load the dataset</span></span>
<span id="cb262-2"><a href="module-3-modeling-part-1.html#cb262-2" tabindex="-1"></a><span class="fu">data</span>(rnaseq)</span>
<span id="cb262-3"><a href="module-3-modeling-part-1.html#cb262-3" tabindex="-1"></a></span>
<span id="cb262-4"><a href="module-3-modeling-part-1.html#cb262-4" tabindex="-1"></a><span class="co"># look at description of the dataset</span></span>
<span id="cb262-5"><a href="module-3-modeling-part-1.html#cb262-5" tabindex="-1"></a>?rnaseq</span>
<span id="cb262-6"><a href="module-3-modeling-part-1.html#cb262-6" tabindex="-1"></a></span>
<span id="cb262-7"><a href="module-3-modeling-part-1.html#cb262-7" tabindex="-1"></a><span class="co"># look at first 6 rows</span></span>
<span id="cb262-8"><a href="module-3-modeling-part-1.html#cb262-8" tabindex="-1"></a><span class="fu">head</span>(rnaseq)</span></code></pre></div>
<pre><code>##    X1  X2 X3  X4  X5 X6 totalReads treatment gender age
## 1 295  65 19 114  54 20   28317494         0      0  57
## 2 213 126 12  96  50  4   20015549         0      0  67
## 3 322 147 23 245  42 19   35318251         0      1  37
## 4 184  35  9 134  74  8   20421437         0      0  59
## 5 376 104  4  74  39  7   21940693         0      1  57
## 6 490  97 18 282 159 24   48645477         0      1  50</code></pre>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="module-3-modeling-part-1.html#cb264-1" tabindex="-1"></a><span class="co"># look at summary of variables</span></span>
<span id="cb264-2"><a href="module-3-modeling-part-1.html#cb264-2" tabindex="-1"></a><span class="fu">summary</span>(rnaseq)</span></code></pre></div>
<pre><code>##        X1              X2               X3              X4       
##  Min.   : 29.0   Min.   : 15.00   Min.   :  3.0   Min.   : 54.0  
##  1st Qu.:132.8   1st Qu.: 43.00   1st Qu.: 15.0   1st Qu.:138.8  
##  Median :191.0   Median : 66.00   Median : 32.0   Median :201.5  
##  Mean   :215.6   Mean   : 73.72   Mean   : 68.1   Mean   :242.6  
##  3rd Qu.:275.8   3rd Qu.: 96.25   3rd Qu.:109.2   3rd Qu.:320.5  
##  Max.   :711.0   Max.   :186.00   Max.   :259.0   Max.   :684.0  
##        X5               X6          totalReads         treatment  
##  Min.   : 18.00   Min.   : 2.00   Min.   :20015549   Min.   :0.0  
##  1st Qu.: 40.00   1st Qu.: 8.00   1st Qu.:26675268   1st Qu.:0.0  
##  Median : 58.00   Median :12.50   Median :33177688   Median :0.5  
##  Mean   : 65.25   Mean   :14.69   Mean   :33748425   Mean   :0.5  
##  3rd Qu.: 81.00   3rd Qu.:19.00   3rd Qu.:40486800   3rd Qu.:1.0  
##  Max.   :183.00   Max.   :41.00   Max.   :49813815   Max.   :1.0  
##      gender           age       
##  Min.   :0.000   Min.   :26.00  
##  1st Qu.:0.000   1st Qu.:43.00  
##  Median :1.000   Median :50.00  
##  Mean   :0.525   Mean   :49.69  
##  3rd Qu.:1.000   3rd Qu.:56.25  
##  Max.   :1.000   Max.   :77.00</code></pre>
<ol style="list-style-type: decimal">
<li>Look up the function MGLMreg to learn about it. What distributions can you fit with this function?</li>
</ol>
<p>Multinomial (dist=“MN”), Dirichlet-multinomial (dist=“DM”), generalized Dirichlet-multinomial (dist=“GDM”), and negative multinomial (dist=“NM”).</p>
<ol start="2" style="list-style-type: decimal">
<li>Let’s fit a multinomial regression to the 6 gene expression columns with gender, age, and treatment as our covariates. We will include log(totalreads) as well.</li>
</ol>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="module-3-modeling-part-1.html#cb266-1" tabindex="-1"></a><span class="co"># the multinomial outcome model would assume NO overdispersion:</span></span>
<span id="cb266-2"><a href="module-3-modeling-part-1.html#cb266-2" tabindex="-1"></a>mnreg <span class="ot">&lt;-</span> <span class="fu">MGLMreg</span>(<span class="at">formula =</span> <span class="fu">cbind</span>(X1, X2, X3, X4, X5, X6) <span class="sc">~</span> <span class="fu">log</span>(totalReads) <span class="sc">+</span></span>
<span id="cb266-3"><a href="module-3-modeling-part-1.html#cb266-3" tabindex="-1"></a>                                 treatment <span class="sc">+</span> age <span class="sc">+</span> gender, <span class="at">data =</span> rnaseq, <span class="at">dist =</span> <span class="st">&quot;MN&quot;</span>) </span>
<span id="cb266-4"><a href="module-3-modeling-part-1.html#cb266-4" tabindex="-1"></a></span>
<span id="cb266-5"><a href="module-3-modeling-part-1.html#cb266-5" tabindex="-1"></a><span class="fu">print</span>(mnreg)</span></code></pre></div>
<pre><code>## Call: MGLMreg(formula = cbind(X1, X2, X3, X4, X5, X6) ~ log(totalReads) + 
##     treatment + age + gender, data = rnaseq, dist = &quot;MN&quot;)
## 
## Coefficients:
##             X1        X2        X3        X4        X5
## [1,]  4.942732  5.009649 -3.792216  4.435434  4.027689
## [2,] -0.112845 -0.170222  0.219277 -0.107260 -0.120928
## [3,] -0.022655 -0.043099  2.745277  1.405742  0.092246
## [4,] -0.006187 -0.009709 -0.005907 -0.010945 -0.009599
## [5,]  0.032676  0.100389  0.020663  0.103859  0.009514
## 
## Hypothesis test: 
##                  wald value    Pr(&gt;wald)
## (Intercept)       144.88789 1.634268e-29
## log(totalReads)    69.92572 1.061922e-13
## treatment       18364.13260 0.000000e+00
## age                79.91023 8.762650e-16
## gender             52.33670 4.601575e-10
## 
## Distribution: Multinomial
## Log-likelihood: -7506.393
## BIC: 15145.24
## AIC: 15062.79
## Iterations: 6</code></pre>
<p>Each row of the coefficient matrix in this output corresponds to our covariates (intercept, log(totalReads), treatment, age, gender) and each column corresponds to our outcomes (X1,…,X6). For example, the coefficient in the first row and first column (4.943) estimates the intercept of X1 which gives the log of baseline expression levels. The second row/first column coefficient (-0.113) gives the estimated association between log(totalReads) and gene expression of X1.</p>
<ol start="3" style="list-style-type: decimal">
<li>Given that this dataset is overdispersed, it is likely that we will need to use a different distribution to accommodate this. Choose one of the distributions available other than “MN”</li>
</ol>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="module-3-modeling-part-1.html#cb268-1" tabindex="-1"></a><span class="co"># we can change &quot;MN&quot; in the previous code to &quot;DM&quot; to fit a Dirichlet-multinomial regression</span></span>
<span id="cb268-2"><a href="module-3-modeling-part-1.html#cb268-2" tabindex="-1"></a>dmreg <span class="ot">&lt;-</span> <span class="fu">MGLMreg</span>(<span class="at">formula =</span> <span class="fu">cbind</span>(X1, X2, X3, X4, X5, X6) <span class="sc">~</span> <span class="fu">log</span>(totalReads) <span class="sc">+</span></span>
<span id="cb268-3"><a href="module-3-modeling-part-1.html#cb268-3" tabindex="-1"></a>                   treatment <span class="sc">+</span> age <span class="sc">+</span> gender, <span class="at">data =</span> rnaseq, <span class="at">dist =</span> <span class="st">&quot;DM&quot;</span>) </span>
<span id="cb268-4"><a href="module-3-modeling-part-1.html#cb268-4" tabindex="-1"></a><span class="fu">print</span>(dmreg)</span></code></pre></div>
<pre><code>## Call: MGLMreg(formula = cbind(X1, X2, X3, X4, X5, X6) ~ log(totalReads) + 
##     treatment + age + gender, data = rnaseq, dist = &quot;DM&quot;)
## 
## Coefficients:
##             X1        X2        X3        X4        X5        X6
## [1,] -0.895850 -1.096921 -8.997414 -1.736871 -1.774227 -5.646822
## [2,]  0.221988  0.186919  0.536572  0.252679  0.216672  0.347271
## [3,] -0.679291 -0.686881  1.835585  0.707954 -0.546469 -0.543134
## [4,]  0.010245  0.005227  0.009134  0.004252  0.006090  0.011642
## [5,] -0.026177  0.040244 -0.052842  0.023178 -0.058339 -0.039139
## 
## Hypothesis test: 
##                  wald value  Pr(&gt;wald)
## (Intercept)       14.579069 0.02379579
## log(totalReads)    8.502549 0.20354699
## treatment       1851.437449 0.00000000
## age               13.131512 0.04099442
## gender             4.133364 0.65863419
## 
## Distribution: Dirichlet Multinomial
## Log-likelihood: -4386.941
## BIC: 8932.831
## AIC: 8833.882
## Iterations: 9</code></pre>
<p>If we wanted to know which model provided a better fit, we could compare their information criterion. For instance, the Bayesian Information Criterion (BIC) is designed to balance model complexity and model fit. A lower BIC indicates a better fitting model. Let’s compare the BIC of our two models fit using the BIC() function in R. Which model fits better for this data?</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="module-3-modeling-part-1.html#cb270-1" tabindex="-1"></a><span class="fu">BIC</span>(dmreg)</span></code></pre></div>
<pre><code>## [1] 8932.831</code></pre>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="module-3-modeling-part-1.html#cb272-1" tabindex="-1"></a><span class="fu">BIC</span>(mnreg)</span></code></pre></div>
<pre><code>## [1] 15145.24</code></pre>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-2-dealing-with-missingness.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-4-modeling-part-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/cbw-dev/STAT_2024/blob/main/030-module-3.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
